% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/XGR.plot_peaks.R
\name{XGR.plot_peaks}
\alias{XGR.plot_peaks}
\title{Plot XGR peaks}
\usage{
XGR.plot_peaks(
  gr.lib,
  subset_DT,
  fill_var = "Assay",
  facet_var = "Source",
  geom = "density",
  locus = NULL,
  adjust = 0.2,
  show_plot = TRUE,
  show.legend = TRUE,
  as.ggplot = TRUE,
  trim_xlims = FALSE
)
}
\arguments{
\item{gr.lib}{\code{GRanges} object of annotations.}

\item{subset_DT}{Data.frame with at least the following columns:
\describe{
\item{SNP}{SNP RSID}
\item{CHR}{chromosome}
\item{POS}{position}
}}

\item{geom}{Plot type ("density", or "histogram").}

\item{locus}{Locus name (\emph{optional}).}

\item{adjust}{The granularity of the peaks.}

\item{show_plot}{Print the plot.}
}
\value{
\code{ggbio} track plot.
}
\description{
Plots the distribution of annotations across a genomic region (x-axis).
}
\examples{
\dontrun{
gr.lib <- XGR.download_and_standardize(
    c("ENCODE_DNaseI_ClusteredV3_CellTypes"),
    finemap_dat = echodata::BST1
)
gr.filt <- XGR.filter_sources(gr.lib = gr.lib, n_top_sources = 5)
gr.filt <- XGR.filter_assays(gr.lib = gr.filt, n_top_assays = 5)
xgr.track <- XGR.plot_peaks(
    gr.lib = gr.filt,
    subset_DT = echodata::BST1,
    fill_var = "Assay",
    facet_var = "Source"
)
}
}
\seealso{
Other XGR: 
\code{\link{XGR.download_and_standardize}()},
\code{\link{XGR.enrichment_bootstrap}()},
\code{\link{XGR.enrichment_plot}()},
\code{\link{XGR.enrichment}()},
\code{\link{XGR.filter_assays}()},
\code{\link{XGR.filter_sources}()},
\code{\link{XGR.import_annotations}()},
\code{\link{XGR.iterate_enrichment}()},
\code{\link{XGR.iterate_overlap}()},
\code{\link{XGR.merge_and_process}()},
\code{\link{XGR.parse_metadata}()},
\code{\link{XGR.plot_enrichment}()},
\code{\link{XGR.prepare_foreground_background}()},
\code{\link{XGR.sep_handler}()},
\code{\link{granges_to_bed}()}
}
\concept{XGR}
\keyword{internal}
