<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.10" id="echoannot-09910">echoannot 0.99.10<a class="anchor" aria-label="anchor" href="#echoannot-09910"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-10">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-10"></a></h3>
<ul><li>
<code>annotation_file_name</code>:
<ul><li>New export function.<br></li>
</ul></li>
<li>
<code>ROADMAP_merge_and_process_grl</code> –&gt; <code>ROADMAP_merge_and_process</code>
</li>
<li>
<code>ROADMAP_query</code>:
<ul><li>Embed <code>ROADMAP_merge_and_process</code> as an option.<br></li>
<li>Make <code>save_path</code> more unique by adding params to name.</li>
</ul></li>
<li>
<code>ROADMAP_tabix</code>:
<ul><li>Can now find a read in existing files of the same.</li>
</ul></li>
<li>
<code>XGR_query</code> –&gt; <code>XGR_query</code>:
<ul><li>Embed n_top as an option, passed to <code>XGR_filter_assays</code> and <code>XGR_filter_sources</code>.<br></li>
</ul></li>
<li>
<code>xgr_query</code> –&gt; <code>xgr_example</code> (to avoid conflict with func name)</li>
<li>Move functions to <code>echoplot</code>:
<ul><li>
<code>ROADMAP_plot</code> and plotting subfunctions.<br></li>
<li>
<code>XGR_plot</code> and plotting subfunctions.<br></li>
</ul></li>
<li>Move functions to <code>downloadR</code>:
<ul><li><code>zenodo_upload</code></li>
<li>
<code>zenodo_list</code><br></li>
</ul></li>
<li>Remove Imports:
<ul><li><code>scales</code></li>
<li><code>piggyback</code></li>
<li><code>RColorBrewer</code></li>
<li><code>zen4R</code></li>
</ul></li>
<li>
<code>test_enrichment</code>:
<ul><li>Can convert data.tables to GRL too.</li>
<li>Pass up args from <code>PermTest</code>: <code>min.parallel</code>,<code>force.parallel</code>
</li>
<li>Allow users to set seed.</li>
</ul></li>
<li>Move <code>zenodo_list</code> to <code>downloadR</code>
</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-10">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-10"></a></h3>
<ul><li>
<code>ROADMAP_query</code>: <code>rtracklayer::import</code> has some bugs that prevent it from importing certain bed.gz files. So implemented a workaround using custom functions instead.</li>
<li>Reduce vignette sizes to &lt;5Mb.</li>
<li>Fix <code>super_summary_plot</code> unit tests.</li>
<li>Fix GHA: <a href="https://github.com/master" class="external-link">@master</a> –&gt; <a href="https://github.com/v2" class="external-link">@v2</a><br></li>
<li>
<code>NOTT2019_plac_seq_plot</code>:
<ul><li>Get rid of hjust warning.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.9" id="echoannot-0999">echoannot 0.99.9<a class="anchor" aria-label="anchor" href="#echoannot-0999"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-9">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-9"></a></h3>
<ul><li>Added <code>MOTIFBREAKR</code> suite of functions.
<ul><li>
<code>MOTIFBREAKR</code>,<code>MOTIFBREAKR_calc_pvals</code>,<code>MOTIFBREAKR_filter</code>, <code>MOTIFBREAKR_plot</code>,<code>MOTIFBREAKR_summarize</code>
</li>
<li>Fully documented.</li>
<li>First attempts to get parallelization working.<br></li>
<li>Passes up other args from <code>motifbreakR</code> core functions.</li>
<li>Saves motif plots as PDFs.</li>
</ul></li>
<li>Added <code>get_bpparam</code> function.</li>
<li>Added <code>filter_chromatin_states</code> function (internal).</li>
<li>Move to <code>echodata</code> package:
<ul><li><code>get_SNPgroup_counts</code></li>
<li><code>results_report</code></li>
<li>
<code>merge_finemapping_results</code>: and split annotation portions into new <code>annotate_snps</code>.</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-9">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-9"></a></h3>
<ul><li>
<code>get_SNPgroup_counts</code>:
<ul><li>Get rid of <code>.dots</code> and <code>.groups</code> warnings.</li>
</ul></li>
<li>Update main vignette to reflect all changes.</li>
<li>
<code>annotate_snps</code>: Fix subfunctions (there’s been some updates to tool outputs).</li>
<li>Fix <code>ROADMAP_query</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.8" id="echoannot-0998">echoannot 0.99.8<a class="anchor" aria-label="anchor" href="#echoannot-0998"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-8">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-8"></a></h3>
<ul><li>Split <code>peakyfinders</code> from <code>echoannot</code>.</li>
<li>
<code>cell_type_specifity</code>: Derive a EWCE-inspired specificity score for each cell-type and show in plot.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-8">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-8"></a></h3>
<ul><li>
<code>merged_finemap_results</code>: Fix overwriting <code>multifinemap_pattern</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.7" id="echoannot-0997">echoannot 0.99.7<a class="anchor" aria-label="anchor" href="#echoannot-0997"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-7">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-7"></a></h3>
<ul><li>
<code>import_peaks_geo</code>:
<ul><li>Can now import/call peaks in parallel, by splitting queries across chromosomes.</li>
<li>Can now import &gt;1 peak type at a time, and documents it in “peaktype” col. </li>
</ul></li>
<li>
<code>import_peaks</code>:
<ul><li>Pass up args: <code>regex_queries</code>, <code>split_chromosomes</code>
</li>
</ul></li>
<li>New function: <code>process_ids</code>:
<ul><li>Able to get GSM sample names from GSE project ID.</li>
</ul></li>
<li>Split <code>import_peaks_geo</code> into separate<code>import_peaks_*</code> functions for each file type.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-7">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-7"></a></h3>
<ul><li>When only one peak type is returned from GEO search, prevent <code>mapply</code> from turning the results into a matrix with <code>SIMPLIFY=FALSE</code>.</li>
<li>Return empty <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GenomicRanges::GRanges()</a></code> instead of error/NA/NULL during errors, so you can still merge at the end.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.6" id="echoannot-0996">echoannot 0.99.6<a class="anchor" aria-label="anchor" href="#echoannot-0996"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-6">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-6"></a></h3>
<ul><li>
<code>XGR_query</code>: Set <code>dat=NULL</code> to return genome-wide data.</li>
<li>New vignette: <em>cell_type_specific_epigenomics</em>, with Nott2019 and Corces2020 data.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-6">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-6"></a></h3>
<ul><li>
<code>prepare_highlight_plac_data</code>: Allow to plot overlap with any SNP group (not just consensus SNPs).</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.5" id="echoannot-0995">echoannot 0.99.5<a class="anchor" aria-label="anchor" href="#echoannot-0995"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-5">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-5"></a></h3>
<ul><li>Moved <code>get_CS_counts</code>/<code>get_CS_bins</code> to <code>echodata</code>.<br></li>
<li>Cache remote data resources in <code>echoannot</code>-specific folder.</li>
<li>Add XGR GitHub installation again (CRAN installation not working again?).</li>
<li>Shorten Nott2019 ref links.</li>
<li>Pass all CRAN checks.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-5">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-5"></a></h3>
<ul><li>Added <code>R.utils</code>/<code>tools</code> to <em>Imports</em>.<br></li>
<li>Added <code>GEOquery</code>/<code>regioneR</code>/ <code>BSgenome.Hsapiens.UCSC.hg38</code>/<code>BSgenome.Hsapiens.UCSC.hg19</code> to <em>Suggests</em>.<br></li>
<li>Updated GHA workflow to account for git security changes.</li>
<li>Fixed <code>convert_plots</code> when converting <code>Tracks</code> to list of ggplots.</li>
<li>Fixed <code>merge_celltype_specific_epigenomics</code>
<ul><li>Added caching mechanism.</li>
</ul></li>
<li>Use <code>echodata::&lt;locus&gt;</code> data instead of portal data for <em>Getting started</em> vignette.</li>
<li>Replace <code>file.path</code> with <code>paste</code>.</li>
<li>Replace “echoR” with “echoR_mini”.</li>
<li>Set timeout with multiple approaches: <code>options</code>,<code><a href="https://httr.r-lib.org/reference/timeout.html" class="external-link">httr::timeout</a></code>, <code><a href="https://httr.r-lib.org/reference/config.html" class="external-link">httr::config</a></code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.4" id="echoannot-0994">echoannot 0.99.4<a class="anchor" aria-label="anchor" href="#echoannot-0994"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-4">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-4"></a></h3>
<ul><li>New functions
<ul><li>
<code>import_peaks</code>: Search for peak files in GEO/ENCODE, or compute them from bedGraph files.</li>
<li>
<code>call_peaks</code>: Call peaks from bedGraph files using <code>MACSr</code>.</li>
<li>
<code>test_enrichment</code>: Run permutation enrichment tests.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.3" id="echoannot-0993">echoannot 0.99.3<a class="anchor" aria-label="anchor" href="#echoannot-0993"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-0-99-3">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-3"></a></h3>
<ul><li>Update to match latest <code>echotabix</code> changes.</li>
<li>Update GHA workflow.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.3" id="echoannot-0993-1">echoannot 0.99.3<a class="anchor" aria-label="anchor" href="#echoannot-0993-1"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-3">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-3"></a></h3>
<ul><li>Added function <code>convert_plots</code>.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-3-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-3-1"></a></h3>
<ul><li>Move as many Imports to Suggests as possible.</li>
<li>GHA servers can’t install XGR from CRAN. Switching to <a href="github::hfang-bristol/XGR">GitHub remote</a> installation until they fix this.</li>
<li>Removed docs/ folder.</li>
<li>Reduced vignette size.</li>
<li>Transition all usage of <code>ggplot2</code> to <code>requireNamespace</code> method to reduce NAMESPACE.</li>
<li>Make all new col assignments in <code>GRanges</code> objects use the proper <code>mcols</code> syntax.</li>
<li>Make sure all <code>GRanges</code> in UCSC format in <code>NOTT2019_plac_seq_plot</code>.</li>
<li>Subdivide <code>NOTT2019_plac_seq_plot</code> into subfunctions.</li>
<li>New exports:
<ul><li><code>NOTT2019_get_epigenomic_peaks</code></li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.2" id="echoannot-0992">echoannot 0.99.2<a class="anchor" aria-label="anchor" href="#echoannot-0992"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-2">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-2"></a></h3>
<ul><li>Added a <code>NEWS.md</code> file to track changes to the package.</li>
<li>Replace the following with <code>echoverseTemplate</code> versions:
<ul><li>New hex logo.</li>
<li>GHA</li>
<li>README</li>
<li>Vignette header</li>
</ul></li>
<li>Offload functions to <code>echodata</code>:
<ul><li><code>assign_lead_snp</code></li>
<li><code>find_consensus_snps</code></li>
<li><code>update_cols</code></li>
<li><code>biomart_geneInfo</code></li>
<li><code>find_top_consensus</code></li>
<li><code>dt_to_granges</code></li>
<li><code>is_granges</code></li>
<li><code>granges_to_bed</code></li>
<li><code>reassign_lead_snps</code></li>
</ul></li>
<li>Renamed all functions to exclude “.”:
<ul><li>“ROADMAP.” –&gt; “ROADMAP_”</li>
<li>“NOTT_2019.” –&gt; “NOTT2019_”</li>
<li>etc.</li>
</ul></li>
<li>Added <code>xgr_example</code> example data.</li>
<li>Added unit tests for <code>super_summary_plot</code> and its subfunctions.</li>
<li>Offloaded all liftover functions to <code>echotabix</code>
</li>
<li>Made new wrapper functions to simplify importing/plotting pipelines:
<ul><li><code>XGR_plot</code></li>
<li><code>ROADMAP_plot</code></li>
</ul></li>
</ul></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

