---
title: "Cell-type-specific epigenomics" 
author: "<h4>Author: <i>Brian M. Schilder</i></h4>" 
date: "<h4>Updated: <i>`r format( Sys.Date(), '%b-%d-%Y')`</i></h4>"
output:
  BiocStyle::html_document
vignette: >
    %\VignetteIndexEntry{echoplot} 
    %\usepackage[utf8]{inputenc}
    %\VignetteEngine{knitr::rmarkdown}
---

```{r setup, echo=FALSE, include=FALSE}
pkg <- read.dcf("../DESCRIPTION", fields = "Package")[1]
library(pkg, character.only = TRUE)
```

```R
library(`r pkg`)
```

## Nott2019

`echoannot` includes data generated by
["Nott2019"](https://doi.org/10.1126/science.aay0793):  

> Nott A, Holtman IR, Coufal NG, ... Glass CK. Brain cell type-specific enhancer-promoter interactome maps and disease-risk association. Science. 2019 Nov 29;366(6469):1134-1139. doi: 10.1126/science.aay0793. Epub 2019 Nov 14. PMID: 31727856; PMCID: PMC7028213.

### Import data 

```{r}
superenhancers <- echoannot::get_NOTT2019_superenhancer_interactome()
enhancers_promoters <- echoannot::NOTT2019_get_promoter_interactome_data()
peaks <- echoannot::NOTT2019_get_epigenomic_peaks() 
```


### Plot

```{r}
dat <- echodata::BST1
histo_out <- echoannot::NOTT2019_epigenomic_histograms(dat = dat) 
```

In addition to the plot object, 
tables of both raw read ranges and called peaks are included in the output list.

```{r}
knitr::kable(head(histo_out$data$raw))
knitr::kable(head(histo_out$data$peaks))
```

 
## Corces2020

`echoannot` also includes data generated by
["Corces2019"](https://doi.org/10.1038/s41588-020-00721-x):

> Corces, M.R., Shcherbina, A., Kundu, S. et al. Single-cell epigenomic analyses implicate candidate causal variants at inherited risk loci for Alzheimer’s and Parkinson’s diseases. Nat Genet 52, 1158–1168 (2020). https://doi.org/10.1038/s41588-020-00721-x 

### Import data 

```{r}
bulkATACseq_peaks <- echoannot::get_CORCES2020_bulkATACseq_peaks()
cicero_coaccessibility <- echoannot::get_CORCES2020_cicero_coaccessibility()
hichip_fithichip_loop_calls <- echoannot::get_CORCES2020_hichip_fithichip_loop_calls()
scATACseq_celltype_peaks <- echoannot::get_CORCES2020_scATACseq_celltype_peaks()
scATACseq_peaks <- echoannot::get_CORCES2020_scATACseq_peaks()
```

### Plot

```{r} 
peak_dat <- echoannot::granges_overlap(
    dat1 = dat,  
    chrom_col.1 = "CHR",
    start_col.1 = "POS",
    dat2 = scATACseq_celltype_peaks, 
    chrom_col.2 = "hg38_Chromosome",
    start_col.2 = "hg38_Start",
    end_col.2 = "hg38_Stop")
ggbio::autoplot(peak_dat, 
                ggplot2::aes(y=ExcitatoryNeurons, color=Effect))

```


# Session Info 

<details> 

```{r Session Info}
utils::sessionInfo()
```

</details>  

