<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get overlap between datatable of SNPs and scATAC peaks — CORCES2020_get_ATAC_peak_overlap • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get overlap between datatable of SNPs and scATAC peaks — CORCES2020_get_ATAC_peak_overlap"><meta property="og:description" content="Can optionally add Cicero coaccessibility scores,
which are also derived from scATAC-seq data."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get overlap between datatable of SNPs and scATAC peaks</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/R/CORCES2020_get_ATAC_peak_overlap.R" class="external-link"><code>R/CORCES2020_get_ATAC_peak_overlap.R</code></a></small>
    <div class="hidden name"><code>CORCES2020_get_ATAC_peak_overlap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Can optionally add <code>Cicero</code> coaccessibility scores,
which are also derived from scATAC-seq data.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">CORCES2020_get_ATAC_peak_overlap</span><span class="op">(</span></span>
<span>  <span class="va">query_dat</span>,</span>
<span>  FDR_filter <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  add_cicero <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cell_type_specific <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://doi.org/10.1038/s41588-020-00721-x" class="external-link">https://doi.org/10.1038/s41588-020-00721-x</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query_dat</dt>
<dd><p>Genomic summary statistics data to query with.</p></dd>


<dt>FDR_filter</dt>
<dd><p>Correct p-value threshold.</p></dd>


<dt>add_cicero</dt>
<dd><p>Whether to include 
<a href="https://www.bioconductor.org/packages/release/bioc/html/cicero.html" class="external-link">
cicero</a> results as well.</p></dd>


<dt>cell_type_specific</dt>
<dd><p>Whether to use bulk or cell-type-specific data.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other CORCES2020: 
<code><a href="CORCES2020_get_hichip_fithichip_overlap.html">CORCES2020_get_hichip_fithichip_overlap</a>()</code>,
<code><a href="CORCES2020_prepare_bulkATAC_peak_overlap.html">CORCES2020_prepare_bulkATAC_peak_overlap</a>()</code>,
<code><a href="CORCES2020_prepare_scATAC_peak_overlap.html">CORCES2020_prepare_scATAC_peak_overlap</a>()</code>,
<code><a href="get_CORCES2020_bulkATACseq_peaks.html">get_CORCES2020_bulkATACseq_peaks</a>()</code>,
<code><a href="get_CORCES2020_cicero_coaccessibility.html">get_CORCES2020_cicero_coaccessibility</a>()</code>,
<code><a href="get_CORCES2020_hichip_fithichip_loop_calls.html">get_CORCES2020_hichip_fithichip_loop_calls</a>()</code>,
<code><a href="get_CORCES2020_scATACseq_celltype_peaks.html">get_CORCES2020_scATACseq_celltype_peaks</a>()</code>,
<code><a href="get_CORCES2020_scATACseq_peaks.html">get_CORCES2020_scATACseq_peaks</a>()</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">query_dat</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1.html" class="external-link">BST1</a></span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>,<span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">gr.hits</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu">CORCES2020_get_ATAC_peak_overlap</span><span class="op">(</span>query_dat <span class="op">=</span> <span class="va">query_dat</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CORCES2020:: Extracting overlapping cell-type-specific scATAC-seq peaks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + CORCES2020:: No Cicero hits found.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

