<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Query IMPACT annotations — IMPACT_query • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Query IMPACT annotations — IMPACT_query"><meta property="og:description" content="Query annotations/LD-scores generated by
IMPACT 
(Inference and Modeling of Phenotype-related ACtive Transcription),
IMPACT predicts transcription factor (TF) binding at a motif site by 
learning the epigenomic profiles at those sites 
(primarily ENCODE).
All data are aligned to the hg19 genome build.
All data has also been 
reformatted to tabix indexed files and uploaded to Zenodo 
here
 to allow for rapid querying."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Query IMPACT annotations</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/R/IMPACT_query.R" class="external-link"><code>R/IMPACT_query.R</code></a></small>
    <div class="hidden name"><code>IMPACT_query.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Query annotations/LD-scores generated by
<a href="https://github.com/immunogenomics/IMPACT" class="external-link">IMPACT</a> 
(Inference and Modeling of Phenotype-related ACtive Transcription),
IMPACT predicts transcription factor (TF) binding at a motif site by 
learning the epigenomic profiles at those sites 
(primarily <a href="https://www.encodeproject.org/" class="external-link">ENCODE</a>).
All data are aligned to the <em>hg19</em> genome build.
All data has also been 
reformatted to tabix indexed files and uploaded to Zenodo 
<a href="https://doi.org/10.5281/zenodo.7062238" class="external-link">here</a>
 to allow for rapid querying.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">IMPACT_query</span><span class="op">(</span></span>
<span>  <span class="va">query_dat</span>,</span>
<span>  types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"annot"</span>, <span class="st">"ldscore"</span><span class="op">)</span>,</span>
<span>  populations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"EAS"</span>, <span class="st">"EUR"</span><span class="op">)</span>,</span>
<span>  query_genome <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>  target_genome <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>  overlapping_only <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  output_format <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"wide"</span>, <span class="st">"long"</span>, <span class="st">"list"</span><span class="op">)</span>,</span>
<span>  add_metadata <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  conda_env <span class="op">=</span> <span class="st">"echoR_mini"</span>,</span>
<span>  nThread <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query_dat</dt>
<dd><p>Variant-level summary statistics.</p></dd>


<dt>types</dt>
<dd><p>File types to include.</p></dd>


<dt>populations</dt>
<dd><p>Population ancestries to include 
("EAS" = East Asian; "EUR" = European).</p></dd>


<dt>query_genome</dt>
<dd><p>Genome build that the <code>query_granges</code> is aligned to.</p></dd>


<dt>target_genome</dt>
<dd><p>Genome build of the VCF file.</p></dd>


<dt>overlapping_only</dt>
<dd><p>Remove variants that do not overlap with the 
positions in <code>query_dat</code>.</p></dd>


<dt>output_format</dt>
<dd><p>Output format options:</p><ul><li><p>"wide" : Spread annotation across columns and keep 1 row/SNP.</p></li>
<li><p>"long" : Melt annotation across rows and allow multiple rows/SNP.</p></li>
<li><p>"list" : Do not perform merging of queries and instead return
results as a named list, where the name is 
the file the annotation came from.</p></li>
</ul></dd>


<dt>add_metadata</dt>
<dd><p>Add metadata about each sample 
(<em>Warning</em>: can substantially increase the dataset size).</p></dd>


<dt>conda_env</dt>
<dd><p>Conda environments to search in. 
If <code>NULL</code> (default), will search all conda environments.</p></dd>


<dt>nThread</dt>
<dd><p>Number of threads to use.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A named list or data.table of annotations 
merged with <code>query_dat</code>.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">query_dat</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1.html" class="external-link">BST1</a></span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span>,<span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">annot_dt</span> <span class="op">&lt;-</span> <span class="fu">IMPACT_query</span><span class="op">(</span>query_dat<span class="op">=</span><span class="va">query_dat</span>, populations<span class="op">=</span><span class="st">"EUR"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing GRanges query using min/max ranges across one or more chromosomes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + as_blocks=TRUE: Will query a single range per chromosome that covers all regions requested (plus anything in between).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========= echotabix::query =========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> query_dat is already a GRanges object. Returning directly.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Inferred format: 'table'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Querying tabular tabix file using: Rsamtools.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking query chromosome style is correct.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Chromosome format: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting query results to data.table.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing query: 4:14884541-16649679</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding 'query' column to results.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieved data with 50 rows</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving query ==&gt; /tmp/Rtmp3k34Cf/file5da929a41f76.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========= echotabix::query =========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> query_dat is already a GRanges object. Returning directly.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Inferred format: 'table'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Querying tabular tabix file using: Rsamtools.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking query chromosome style is correct.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Chromosome format: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting query results to data.table.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing query: 4:14884541-16649679</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding 'query' column to results.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieved data with 13 rows</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving query ==&gt; /tmp/Rtmp3k34Cf/file5da91d558a3f.gz</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

