<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Import bigwig files from the UCSC Genome Browser — import_ucsc_bigwigs • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Import bigwig files from the UCSC Genome Browser — import_ucsc_bigwigs"><meta property="og:description" content="Import, preprocess, merge, and save data from 
UCSC Genome Browser tracks."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import bigwig files from the UCSC Genome Browser</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/R/import_ucsc_bigwigs.R" class="external-link"><code>R/import_ucsc_bigwigs.R</code></a></small>
    <div class="hidden name"><code>import_ucsc_bigwigs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Import, preprocess, merge, and save data from 
<a href="https://genome.ucsc.edu/" class="external-link">UCSC Genome Browser</a> tracks.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">import_ucsc_bigwigs</span><span class="op">(</span></span>
<span>  <span class="va">query_dat</span>,</span>
<span>  <span class="va">bigwig_metadata</span>,</span>
<span>  full_data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  xlims <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  force_new <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nThread <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query_dat</dt>
<dd><p><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a> or
 <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a>
containing genomic coordinates to query the UCSC tracks with.</p></dd>


<dt>bigwig_metadata</dt>
<dd><p>Metadata table with at 
least the following two columns:</p><ul><li><p>"name"Unique name of the file.</p></li>
<li><p>"data_link"URL to UCSC genome browser bigwig file.</p></li>
</ul></dd>


<dt>full_data</dt>
<dd><p>Whether to download the full data 
(genomic ranges of all sequence reads) as opposed
to a reduced representation of the data as a single vector 
(i.e. the aggregated reads "score"). 
Setting <code>full_data=TRUE</code> is necessary for creating
 histograms and density plots.</p></dd>


<dt>xlims</dt>
<dd><p>Min/max positions to filter from data after initial query.
This helps to capture genomic ranges that only partially overlap with 
<code>query_dat</code>.</p></dd>


<dt>save_path</dt>
<dd><p>Where to save the processed data.</p></dd>


<dt>force_new</dt>
<dd><p>If a file already exists, download a new one anyway.</p></dd>


<dt>nThread</dt>
<dd><p>Number of threads to parallelise downloads across.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">bigwig_metadata</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="va"><a href="NOTT2019_bigwig_metadata.html">NOTT2019_bigwig_metadata</a></span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">query_dat</span> <span class="op">=</span> <span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1.html" class="external-link">BST1</a></span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">bw.gr</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu">import_ucsc_bigwigs</span><span class="op">(</span>query_dat <span class="op">=</span> <span class="va">query_dat</span>, </span></span>
<span class="r-in"><span>                                        bigwig_metadata <span class="op">=</span> <span class="va">bigwig_metadata</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing GRanges query using min/max ranges across one or more chromosomes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Downloading data from UCSC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing... [1] exvivo_H3K27ac</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving bigwig query ==&gt; /tmp/Rtmp3k34Cf/file5da941675e32</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

