<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="CORCES2020_get_ATAC_peak_overlap.html">CORCES2020_get_ATAC_peak_overlap()</a></code> </p>
        </td>
        <td><p>Get overlap between datatable of SNPs and scATAC peaks</p></td>
      </tr><tr><td>
          <p><code><a href="CORCES2020_get_hichip_fithichip_overlap.html">CORCES2020_get_hichip_fithichip_overlap()</a></code> </p>
        </td>
        <td><p>Get overlap between data table of SNPs and HiChIP_FitHiChIP
coaccessibility anchors</p></td>
      </tr><tr><td>
          <p><code><a href="CS_bin_plot.html">CS_bin_plot()</a></code> </p>
        </td>
        <td><p>Plot CS bin counts</p></td>
      </tr><tr><td>
          <p><code><a href="CS_counts_plot.html">CS_counts_plot()</a></code> </p>
        </td>
        <td><p>Bar plot of tool-specific CS sizes</p></td>
      </tr><tr><td>
          <p><code><a href="IMPACT_files.html">IMPACT_files</a></code> </p>
        </td>
        <td><p>IMPACT files</p></td>
      </tr><tr><td>
          <p><code><a href="IMPACT_query.html">IMPACT_query()</a></code> </p>
        </td>
        <td><p>Query IMPACT annotations</p></td>
      </tr><tr><td>
          <p><code><a href="MOTIFBREAKR.html">MOTIFBREAKR()</a></code> </p>
        </td>
        <td><p>Run <span class="pkg">motifbreakR</span></p></td>
      </tr><tr><td>
          <p><code><a href="MOTIFBREAKR_calc_pvals.html">MOTIFBREAKR_calc_pvals()</a></code> </p>
        </td>
        <td><p>Calculate <span class="pkg">motifbreakR</span> p-values</p></td>
      </tr><tr><td>
          <p><code><a href="MOTIFBREAKR_filter.html">MOTIFBREAKR_filter()</a></code> </p>
        </td>
        <td><p>Merge and filter <span class="pkg">motifbreakR</span> + <span class="pkg">echolocatoR</span> results</p></td>
      </tr><tr><td>
          <p><code><a href="MOTIFBREAKR_plot.html">MOTIFBREAKR_plot()</a></code> </p>
        </td>
        <td><p>Plot <span class="pkg">motifbreakR</span> results</p></td>
      </tr><tr><td>
          <p><code><a href="MOTIFBREAKR_summarize.html">MOTIFBREAKR_summarize()</a></code> </p>
        </td>
        <td><p>Summarize <span class="pkg">motifbreakR</span> + <span class="pkg">echolocatoR</span> results</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT2019_bigwig_metadata.html">NOTT2019_bigwig_metadata</a></code> </p>
        </td>
        <td><p>Metadata and links to data</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT2019_epigenomic_histograms.html">NOTT2019_epigenomic_histograms()</a></code> </p>
        </td>
        <td><p>Plot brain cell-specific epigenomic data</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT2019_get_epigenomic_peaks.html">NOTT2019_get_epigenomic_peaks()</a></code> </p>
        </td>
        <td><p>Download cell type-specific epigenomic peaks</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT2019_get_interactome.html">NOTT2019_get_interactome()</a></code> </p>
        </td>
        <td><p>Import cell type-specific interactomes</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT2019_get_promoter_interactome_data.html">NOTT2019_get_promoter_interactome_data()</a></code> </p>
        </td>
        <td><p>Get cell type-specific promoter/emhancer/interactome data</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT2019_get_regulatory_regions.html">NOTT2019_get_regulatory_regions()</a></code> </p>
        </td>
        <td><p>Get regulatory regions: Nott2019</p></td>
      </tr><tr><td>
          <p><code><a href="NOTT2019_plac_seq_plot.html">NOTT2019_plac_seq_plot()</a></code> </p>
        </td>
        <td><p>Plot brain cell-specific interactome data</p></td>
      </tr><tr><td>
          <p><code><a href="ROADMAP_construct_reference.html">ROADMAP_construct_reference()</a></code> </p>
        </td>
        <td><p>Gather Roadmap annotation metadata</p></td>
      </tr><tr><td>
          <p><code><a href="ROADMAP_query.html">ROADMAP_query()</a></code> </p>
        </td>
        <td><p>Query Roadmap 
 
Query Roadmap annotations using a set of genomic coordinates.</p></td>
      </tr><tr><td>
          <p><code><a href="XGR_enrichment.html">XGR_enrichment()</a></code> </p>
        </td>
        <td><p>XGR enrichment</p></td>
      </tr><tr><td>
          <p><code><a href="XGR_enrichment_bootstrap.html">XGR_enrichment_bootstrap()</a></code> </p>
        </td>
        <td><p>XGR enrichment (bootstrapped)</p></td>
      </tr><tr><td>
          <p><code><a href="XGR_filter_assays.html">XGR_filter_assays()</a></code> </p>
        </td>
        <td><p>Filter assays</p></td>
      </tr><tr><td>
          <p><code><a href="XGR_filter_sources.html">XGR_filter_sources()</a></code> </p>
        </td>
        <td><p>Filter sources</p></td>
      </tr><tr><td>
          <p><code><a href="add_mb.html">add_mb()</a></code> </p>
        </td>
        <td><p>Add Mb</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_missense.html">annotate_missense()</a></code> </p>
        </td>
        <td><p>Annotate any missense variants</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_snps.html">annotate_snps()</a></code> </p>
        </td>
        <td><p>Annotate merged fine-mapping results from all loci</p></td>
      </tr><tr><td>
          <p><code><a href="annotation_file_name.html">annotation_file_name()</a></code> </p>
        </td>
        <td><p>Annotation file name</p></td>
      </tr><tr><td>
          <p><code><a href="convert_plots.html">convert_plots()</a></code> </p>
        </td>
        <td><p>Convert plots to various formats</p></td>
      </tr><tr><td>
          <p><code><a href="get_CORCES2020_bulkATACseq_peaks.html">get_CORCES2020_bulkATACseq_peaks()</a></code> </p>
        </td>
        <td><p>bulkATACseq peaks from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="get_CORCES2020_cicero_coaccessibility.html">get_CORCES2020_cicero_coaccessibility()</a></code> </p>
        </td>
        <td><p>Cicero_coaccessibility from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="get_CORCES2020_hichip_fithichip_loop_calls.html">get_CORCES2020_hichip_fithichip_loop_calls()</a></code> </p>
        </td>
        <td><p>FitHiChIP loop calls from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="get_CORCES2020_scATACseq_celltype_peaks.html">get_CORCES2020_scATACseq_celltype_peaks()</a></code> </p>
        </td>
        <td><p>scATACseq cell type-specific peaks from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="get_CORCES2020_scATACseq_peaks.html">get_CORCES2020_scATACseq_peaks()</a></code> </p>
        </td>
        <td><p>scATACseq peaks from human brain tissue</p></td>
      </tr><tr><td>
          <p><code><a href="get_NOTT2019_superenhancer_interactome.html">get_NOTT2019_superenhancer_interactome()</a></code> </p>
        </td>
        <td><p>Brain cell type-specific interactomes with superenhancers</p></td>
      </tr><tr><td>
          <p><code><a href="get_window_limits.html">get_window_limits()</a></code> </p>
        </td>
        <td><p>Get widow limits</p></td>
      </tr><tr><td>
          <p><code><a href="granges_overlap.html">granges_overlap()</a></code> </p>
        </td>
        <td><p>Find GenomicRanges overlap
 
Find overlap genomic position overlap between two
<a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a> objects.</p></td>
      </tr><tr><td>
          <p><code><a href="import_ucsc_bigwigs.html">import_ucsc_bigwigs()</a></code> </p>
        </td>
        <td><p>Import bigwig files from the UCSC Genome Browser</p></td>
      </tr><tr><td>
          <p><code><a href="merge_celltype_specific_epigenomics.html">merge_celltype_specific_epigenomics()</a></code> </p>
        </td>
        <td><p>Merge all cell-type-specific epigenomics</p></td>
      </tr><tr><td>
          <p><code><a href="plot_dataset_overlap.html">plot_dataset_overlap()</a></code> </p>
        </td>
        <td><p>Plot inter-study SNP overlap</p></td>
      </tr><tr><td>
          <p><code><a href="plot_missense.html">plot_missense()</a></code> </p>
        </td>
        <td><p>Plot any missense variants</p></td>
      </tr><tr><td>
          <p><code><a href="super_summary_plot.html">super_summary_plot()</a></code> </p>
        </td>
        <td><p>Merge all summary plots into one super plot</p></td>
      </tr><tr><td>
          <p><code><a href="test_enrichment.html">test_enrichment()</a></code> </p>
        </td>
        <td><p>Test enrichment</p></td>
      </tr><tr><td>
          <p><code><a href="xgr_example.html">xgr_example</a></code> </p>
        </td>
        <td><p>Example XGR query</p></td>
      </tr><tr><td>
          <p><code><a href="xgr_query.html">XGR_query()</a></code> </p>
        </td>
        <td><p>Download, standardize, and merge XGR annotations</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

