<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Merge and filter motifbreakR + echolocatoR results — MOTIFBREAKR_filter • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Merge and filter motifbreakR + echolocatoR results — MOTIFBREAKR_filter"><meta property="og:description" content="For each SNP we have at least one allele achieving a p-value&amp;lt;1e-4 threshold
 that we required.
The seqMatch column shows what the reference genome sequence 
is at that location,
with the variant position appearing in an uppercase letter.
pctRef and pctAlt display the the score for the motif 
in the sequence as a percentage of the best score that motif 
could achieve on an ideal sequence.
In other words, (scoreVariant−minscorePWM)/(maxscorePWM−minscorePWM).
We can also see the absolute scores for our method in
 scoreRef and scoreAlt and their respective p-values."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Merge and filter <span class="pkg">motifbreakR</span> + <span class="pkg">echolocatoR</span> results</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/R/MOTIFBREAKR_filter.R" class="external-link"><code>R/MOTIFBREAKR_filter.R</code></a></small>
    <div class="hidden name"><code>MOTIFBREAKR_filter.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For each SNP we have at least one allele achieving a p-value&lt;1e-4 threshold
 that we required.
The <em>seqMatch</em> column shows what the reference genome sequence 
is at that location,
with the variant position appearing in an uppercase letter.
<em>pctRef</em> and <em>pctAlt</em> display the the score for the motif 
in the sequence as a percentage of the best score that motif 
could achieve on an ideal sequence.
In other words, <em>(scoreVariant−minscorePWM)/(maxscorePWM−minscorePWM)</em>.
We can also see the absolute scores for our method in
 <em>scoreRef</em> and <em>scoreAlt</em> and their respective p-values.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MOTIFBREAKR_filter</span><span class="op">(</span></span>
<span>  <span class="va">mb_res</span>,</span>
<span>  <span class="va">merged_DT</span>,</span>
<span>  filter_by_locus <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  remove_NA_geneSymbol <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pct_threshold <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pvalue_threshold <span class="op">=</span> <span class="fl">1e-04</span>,</span>
<span>  qvalue_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  effect_strengths <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  snp_filter <span class="op">=</span> <span class="st">"Support&gt;0"</span>,</span>
<span>  top_geneSymbol_hits <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  no_no_loci <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mb_res</dt>
<dd><p>Results generated by <a href="MOTIFBREAKR.html">MOTIFBREAKR</a>, in 
<a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a> format.</p></dd>


<dt>merged_DT</dt>
<dd><p>Table with columns <em>Locus</em> and 
<em>SNP</em> to filter <code>mb_res</code> by.</p></dd>


<dt>filter_by_locus</dt>
<dd><p>Filter <code>mb_res</code> 
to only include SNPs present in a given <em>Locus</em> (e.g. <code>"BST1"</code>).
Set to <code>NULL</code> (default) for not perform any filtering.
Requires <code>merged_DT</code> argument to be supplied.</p></dd>


<dt>remove_NA_geneSymbol</dt>
<dd><p>Remove results where <code>geneSymbol==NA</code>.</p></dd>


<dt>pct_threshold</dt>
<dd><p>Remove rows below the percentage of 
the optimal binding score (PCT) threshold.</p></dd>


<dt>pvalue_threshold</dt>
<dd><p>Remove rows below the raw
significance value (p-value) threshold.</p></dd>


<dt>qvalue_threshold</dt>
<dd><p>Remove rows below the multiple testing-corrected 
significance value (q-value) threshold.</p></dd>


<dt>effect_strengths</dt>
<dd><p>Only include results with certain effect strengths.</p></dd>


<dt>snp_filter</dt>
<dd><p>Condition to filter SNPs by, 
after <code>mb_res</code> and <code>merged_DT</code> have been merged together 
into one table.</p></dd>


<dt>top_geneSymbol_hits</dt>
<dd><p>Only include N top results per gene symbol 
based on absolute <code>risk_score</code>, where N=<code>top_geneSymbol_hits</code>.
If <code>top_geneSymbol_hits=NULL</code>, no such filtering is performed.</p></dd>


<dt>no_no_loci</dt>
<dd><p>Filter out SNPs contained within specific loci
in the <code>merged_DT</code> table.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></p>
 

<p>with the filtered motif disruption results 
after merging with <code>merged_DT</code>.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">merged_DT</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/echodata/man/get_Nalls2019_merged.html" class="external-link">get_Nalls2019_merged</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">mb_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="MOTIFBREAKR.html">MOTIFBREAKR</a></span><span class="op">(</span>rsid_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs11175620"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                      <span class="co"># limit the number of datasets tests </span></span></span>
<span class="r-in"><span>                      <span class="co"># for demonstration purposes only</span></span></span>
<span class="r-in"><span>                      pwmList_max <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>                      calculate_pvals <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> genome_build set to hg19 by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using genome_build hg19</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + MOTIFBREAKR:: Using pre-existing tmp file.</span>
<span class="r-in"><span><span class="va">mb_res_filt</span> <span class="op">&lt;-</span> <span class="fu">MOTIFBREAKR_filter</span><span class="op">(</span>mb_res <span class="op">=</span> <span class="va">mb_res</span>,</span></span>
<span class="r-in"><span>                                  merged_DT <span class="op">=</span> <span class="va">merged_DT</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating 'risk_' columns</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + MOTIFBREAKR 1 SNPs in results.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + MOTIFBREAKR 1 SNPs in results @ `snp_filter='Support&gt;0'`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + MOTIFBREAKR 0 SNPs in results @ `pvalue_threshold=1e-04`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + MOTIFBREAKR 0 SNPs in results @ `qvalue_threshold=0.05`</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

