<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Run motifbreakR — MOTIFBREAKR • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Run motifbreakR — MOTIFBREAKR"><meta property="og:description" content="motifbreakR is a package to predict how much a SNP will disrupt
a transcription factor binding motif (if it falls within one).
Notes:
BSgenomeUsers must manually run library(BSgenome) 
before running any motifbreakR functions 
to successfully use this tool.
threshold=
 If filterp=TRUE, this argument indicates the p-value threshold.
 If filterp=FALSE, this argument instead indicates the pct threshold.

"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run <span class="pkg">motifbreakR</span></h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/R/MOTIFBREAKR.R" class="external-link"><code>R/MOTIFBREAKR.R</code></a></small>
    <div class="hidden name"><code>MOTIFBREAKR.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><span class="pkg">motifbreakR</span> is a package to predict how much a SNP will disrupt
a transcription factor binding motif (if it falls within one).
<em>Notes:</em><br></p><ul><li><p><span class="pkg">BSgenome</span>Users must manually run <code><a href="https://bioconductor.org/packages/BSgenome" class="external-link">library(BSgenome)</a></code> 
before running any <span class="pkg">motifbreakR</span> functions 
to successfully use this tool.</p></li>
<li><p><code>threshold=</code>
 If <code>filterp=TRUE</code>, this argument indicates the p-value threshold.
 If <code>filterp=FALSE</code>, this argument instead indicates the pct threshold.</p></li>
</ul></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MOTIFBREAKR</span><span class="op">(</span></span>
<span>  <span class="va">rsid_list</span>,</span>
<span>  results_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"results"</span><span class="op">)</span>,</span>
<span>  pwmList <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pwmList_max <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genome_build <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  organism <span class="op">=</span> <span class="st">"Hsapiens"</span>,</span>
<span>  threshold <span class="op">=</span> <span class="fl">0.85</span>,</span>
<span>  show.neutral <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"default"</span>,</span>
<span>  calculate_pvals <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  force_new <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  background <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A <span class="op">=</span> <span class="fl">0.25</span>, C <span class="op">=</span> <span class="fl">0.25</span>, G <span class="op">=</span> <span class="fl">0.25</span>, T <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span>,</span>
<span>  granularity <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  nThread <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://pubmed.ncbi.nlm.nih.gov/26272984/" class="external-link">Publication</a>
<a href="https://github.com/Simon-Coetzee/MotifBreakR" class="external-link">GitHub</a>
<a href="http://simon-coetzee.github.io/motifBreakR" class="external-link">Vingette</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>rsid_list</dt>
<dd><p>RSIDs of SNPs to test for motif disruption 
between the reference and alternative alleles..</p></dd>


<dt>results_dir</dt>
<dd><p>Directory where results should be saved 
as a file named:
 <em>&lt;results_dir&gt;/_genome_wide/motifbreakR/motifbreakR_results.rds</em>.
If <code>NULL</code>, results will not be saved to disk.</p></dd>


<dt>pwmList</dt>
<dd><p>An object of class <code>MotifList</code> containing the motifs that
you wish to interrogate</p></dd>


<dt>pwmList_max</dt>
<dd><p>Limit the maximum number of PWM datasets tested 
(e.g. <code>10</code>). If <code>NULL</code>, no limit it set.</p></dd>


<dt>genome_build</dt>
<dd><p>Genome build to use.</p></dd>


<dt>organism</dt>
<dd><p>Only include datasets in the <code>pwmList</code> 
performed in a particular organism.</p></dd>


<dt>threshold</dt>
<dd><p>Numeric; the maximum p-value for a match to be called or a minimum score threshold</p></dd>


<dt>show.neutral</dt>
<dd><p>Logical; include neutral changes in the output</p></dd>


<dt>method</dt>
<dd><p>Character; one of <code>default</code>, <code>log</code>, <code>ic</code>, or <code>notrans</code>; see
details.</p></dd>


<dt>calculate_pvals</dt>
<dd><p>Calculate p-values for all SNPs tested.
<em>WARNING:</em> May take a long time if many SNPs and/or PWM are selected.</p></dd>


<dt>force_new</dt>
<dd><p>If results of the same name already exist, 
overwrite them with new analyses (<code>TRUE</code>). 
Otherwise, import the existing results and skip the analyses
 (default: <code>FALSE</code>).</p></dd>


<dt>background</dt>
<dd><p>Numeric Vector; the background probabilities of the nucleotides</p></dd>


<dt>granularity</dt>
<dd><p>Numeric Vector; the granularity to which to round the PWM,
larger values compromise full accuracy for speed of calculation. A value of
<code>NULL</code> does no rounding.</p></dd>


<dt>nThread</dt>
<dd><p>Number of threads to parallelize analyses across.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Motif disruption predictions in</p>
<p></p>
<p><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a> format.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other motifbreakR: 
<code><a href="MOTIFBREAKR_filter_by_metadata.html">MOTIFBREAKR_filter_by_metadata</a>()</code>,
<code><a href="MOTIFBREAKR_summarize.html">MOTIFBREAKR_summarize</a>()</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/BSgenome" class="external-link">BSgenome</a></span><span class="op">)</span> <span class="co">## &lt;-- IMPORTANT!</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'BiocGenerics'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:stats':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'S4Vectors'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biostrings</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: XVector</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'Biostrings'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     strsplit</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: rtracklayer</span>
<span class="r-in"><span><span class="co">#### Example fine-mapping results ####</span></span></span>
<span class="r-in"><span><span class="va">merged_DT</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/echodata/man/get_Nalls2019_merged.html" class="external-link">get_Nalls2019_merged</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#### Run motif analyses ####</span></span></span>
<span class="r-in"><span><span class="va">mb_res</span> <span class="op">&lt;-</span> <span class="fu">MOTIFBREAKR</span><span class="op">(</span>rsid_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs11175620"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                      <span class="co"># limit the number of datasets tested </span></span></span>
<span class="r-in"><span>                      <span class="co"># for demonstration purposes only</span></span></span>
<span class="r-in"><span>                      pwmList_max <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>                      calculate_pvals <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: motifbreakR</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> genome_build set to hg19 by default.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: SNPlocs.Hsapiens.dbSNP144.GRCh37</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: BSgenome.Hsapiens.UCSC.hg19</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using genome_build hg19</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + MOTIFBREAKR:: Converting SNP list into motifbreakR input format.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> See system.file("LICENSE", package="MotifDb") for use restrictions.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering pwmList to only include organism: Hsapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Only testing the first 4 datasets in pwmList.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + MOTIFBREAKR:: Identifying motifs and predicting disruptions.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> reached end of SNPs list length = 1 with 1 potentially disruptive matches to 1 of 4 motifs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + MOTIFBREAKR:: Saving results ==&gt; /tmp/Rtmp3k34Cf/results/_genome_wide/motifbreakR/motifbreakR_results.rds</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

