<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot brain cell-specific interactome data — NOTT2019_plac_seq_plot • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot brain cell-specific interactome data — NOTT2019_plac_seq_plot"><meta property="og:description" content="Plot brain cell-specific interactome data from
 Nott et al. (2019)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot brain cell-specific interactome data</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/R/NOTT2019_plac_seq_plot.R" class="external-link"><code>R/NOTT2019_plac_seq_plot.R</code></a></small>
    <div class="hidden name"><code>NOTT2019_plac_seq_plot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot brain cell-specific interactome data from
 <a href="https://doi.org/10.1126/science.aay0793" class="external-link">Nott et al. (2019)</a>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">NOTT2019_plac_seq_plot</span><span class="op">(</span></span>
<span>  dat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  locus_dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  return_interaction_track <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  x_limits <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  zoom_window <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  index_SNP <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genomic_units <span class="op">=</span> <span class="st">"POS"</span>,</span>
<span>  color_dict <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>enhancers <span class="op">=</span> <span class="st">"springgreen2"</span>, promoters <span class="op">=</span> <span class="st">"purple"</span>, anchors <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>  highlight_plac <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_regulatory_rects <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_anchors <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  strip.text.y.angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  xtext <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  save_annot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  height <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  dpi <span class="op">=</span> <span class="fl">300</span>,</span>
<span>  return_as <span class="op">=</span> <span class="st">"Tracks"</span>,</span>
<span>  nThread <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://doi.org/10.1126/science.aay0793" class="external-link">Nott et al. (2019)</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dat</dt>
<dd><p>Fine-mapping results data from finemap_loci.</p></dd>


<dt>locus_dir</dt>
<dd><p>Locus-specific directory.</p></dd>


<dt>title</dt>
<dd><p>all title elements: plot, axes, legends (<code><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text()</a></code>;
inherits from <code>text</code>)</p></dd>


<dt>show_plot</dt>
<dd><p>Print plot.</p></dd>


<dt>save_plot</dt>
<dd><p>Whether to save the plot.</p></dd>


<dt>return_interaction_track</dt>
<dd><p>Return only the interaction track 
(before completing the plot and showing it).</p></dd>


<dt>x_limits</dt>
<dd><p>x-axis limits to be applied to all plots
(useful when trying to keep a common coordinate system).</p></dd>


<dt>zoom_window</dt>
<dd><p>Zoom window.</p></dd>


<dt>index_SNP</dt>
<dd><p>Index/lead SNP RSID.</p></dd>


<dt>genomic_units</dt>
<dd><p>Which genomic units to return window limits in.</p></dd>


<dt>color_dict</dt>
<dd><p>Named list of colors for each regulatory element.</p></dd>


<dt>highlight_plac</dt>
<dd><p>Whether to scale opacity of PLAC-seq interactions 
(arches) such that interactions with anchors containing Consensus SNPs 
will be colored darker (Default: <code>TRUE</code>).
If <code>FALSE</code>, will instead apply the same opacity level
 to all interactions.</p></dd>


<dt>show_regulatory_rects</dt>
<dd><p>Show enhancers/promoters as rectangles.</p></dd>


<dt>show_anchors</dt>
<dd><p>Show PLAC-seq anchors.</p></dd>


<dt>strip.text.y.angle</dt>
<dd><p>Angle of the y-axis facet labels.</p></dd>


<dt>xtext</dt>
<dd><p>Whether to include x-axis title and text.</p></dd>


<dt>save_annot</dt>
<dd><p>Save the queried subset of bigwig annotations.</p></dd>


<dt>point_size</dt>
<dd><p>Point size of each SNP in the GWAS/fine-mapping plots.</p></dd>


<dt>height</dt>
<dd><p>height (defaults to the height of current
  plotting window)</p></dd>


<dt>width</dt>
<dd><p>width (defaults to the width of current
  plotting window)</p></dd>


<dt>dpi</dt>
<dd><p>dpi to use for raster graphics</p></dd>


<dt>return_as</dt>
<dd><p>Plot class to convert <code>plot_list</code> to:</p><ul><li><p>"ggplot"<a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></p></li>
<li><p>"ggbio"<a href="https://rdrr.io/pkg/ggbio/man/ggbio-class.html" class="external-link">ggbio</a></p></li>
<li><p>"patchwork"<span class="pkg">patchwork</span></p></li>
<li><p>"Tracks"<a href="https://rdrr.io/pkg/ggbio/man/tracks.html" class="external-link">tracks</a></p></li>
<li><p>NULLReturn original object.</p></li>
</ul></dd>


<dt>nThread</dt>
<dd><p>Number of threads to parallelise downloads across.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other NOTT2019: 
<code><a href="NOTT2019_bigwig_metadata.html">NOTT2019_bigwig_metadata</a></code>,
<code><a href="NOTT2019_epigenomic_histograms.html">NOTT2019_epigenomic_histograms</a>()</code>,
<code><a href="NOTT2019_get_epigenomic_peaks.html">NOTT2019_get_epigenomic_peaks</a>()</code>,
<code><a href="NOTT2019_get_interactions.html">NOTT2019_get_interactions</a>()</code>,
<code><a href="NOTT2019_get_interactome.html">NOTT2019_get_interactome</a>()</code>,
<code><a href="NOTT2019_get_promoter_celltypes.html">NOTT2019_get_promoter_celltypes</a>()</code>,
<code><a href="NOTT2019_get_promoter_interactome_data.html">NOTT2019_get_promoter_interactome_data</a>()</code>,
<code><a href="NOTT2019_get_regulatory_regions.html">NOTT2019_get_regulatory_regions</a>()</code>,
<code><a href="NOTT2019_superenhancers.html">NOTT2019_superenhancers</a>()</code>,
<code><a href="get_NOTT2019_interactome.html">get_NOTT2019_interactome</a>()</code>,
<code><a href="get_NOTT2019_superenhancer_interactome.html">get_NOTT2019_superenhancer_interactome</a>()</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">trks_plus_lines</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu">NOTT2019_plac_seq_plot</span><span class="op">(</span>dat <span class="op">=</span> <span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1.html" class="external-link">BST1</a></span><span class="op">)</span> </span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> NOTT2019:: Creating PLAC-seq interactome plot</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ NOTT2019:: Getting promoter cell-type-specific data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ NOTT2019:: Getting interactome data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ NOTT2019:: Getting regulatory regions data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Astrocyte enhancers ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Astrocyte promoters ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Neuronal enhancers ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Neuronal promoters ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Oligo enhancers ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Oligo promoters ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Microglia enhancers ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Microglia promoters ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ NOTT2019:: Getting interaction anchors data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Microglia interactome ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Neuronal interactome ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Oligo interactome ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 29 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 49 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for highlighting PLAC-seq interactions that overlap with SNP subset: Support&gt;0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Initializing PLAC-seq plot.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ Adding enhancer/promoter rectangles</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating GWAS track.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating fine-map track.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> x_limits will be used to limit the min/max x-axis values for all plots.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting plots to a merged ggbio Tracks object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding vertical track lines for LeadSNP and Consensus_SNP</span>
<span class="r-plt img"><img src="NOTT2019_plac_seq_plot-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

