<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot brain cell-specific epigenomic data — NOTT2019_epigenomic_histograms • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot brain cell-specific epigenomic data — NOTT2019_epigenomic_histograms"><meta property="og:description" content="Brain cell-specific epigenomic data from Nott et al. (2019)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot brain cell-specific epigenomic data</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/R/NOTT2019_epigenomic_histograms.R" class="external-link"><code>R/NOTT2019_epigenomic_histograms.R</code></a></small>
    <div class="hidden name"><code>NOTT2019_epigenomic_histograms.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Brain cell-specific epigenomic data from Nott et al. (2019).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">NOTT2019_epigenomic_histograms</span><span class="op">(</span></span>
<span>  <span class="va">dat</span>,</span>
<span>  bigwig_metadata <span class="op">=</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="va"><a href="NOTT2019_bigwig_metadata.html">NOTT2019_bigwig_metadata</a></span>,</span>
<span>  locus_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  save_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  full_data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  return_assay_track <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  binwidth <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  density_adjust <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  zoom <span class="op">=</span> <span class="st">"1x"</span>,</span>
<span>  strip.text.y.angle <span class="op">=</span> <span class="fl">90</span>,</span>
<span>  xtext <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  geom <span class="op">=</span> <span class="st">"density"</span>,</span>
<span>  plot_formula <span class="op">=</span> <span class="st">"Cell_type ~."</span>,</span>
<span>  fill_var <span class="op">=</span> <span class="st">"Assay"</span>,</span>
<span>  genomic_units <span class="op">=</span> <span class="st">"Mb"</span>,</span>
<span>  as_ggplot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  dpi <span class="op">=</span> <span class="fl">300</span>,</span>
<span>  height <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  nThread <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  save_annot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://doi.org/10.1126/science.aay0793" class="external-link">Nott et al. (2019)</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dat</dt>
<dd><p>Fine-mapping results data from finemap_loci.</p></dd>


<dt>bigwig_metadata</dt>
<dd><p>Metadata table with at 
least the following two columns:</p><ul><li><p>"name"Unique name of the file.</p></li>
<li><p>"data_link"URL to UCSC genome browser bigwig file.</p></li>
</ul></dd>


<dt>locus_dir</dt>
<dd><p>Locus-specific directory.</p></dd>


<dt>show_plot</dt>
<dd><p>Show plot.</p></dd>


<dt>save_plot</dt>
<dd><p>Whether to save the plot.</p></dd>


<dt>full_data</dt>
<dd><p>Whether to download the full data 
(genomic ranges of all sequence reads) as opposed
to a reduced representation of the data as a single vector 
(i.e. the aggregated reads "score"). 
Setting <code>full_data=TRUE</code> is necessary for creating
 histograms and density plots.</p></dd>


<dt>return_assay_track</dt>
<dd><p>Return only the assay track 
(before adding the rest of the tracks and showing the plot).</p></dd>


<dt>binwidth</dt>
<dd><p>width of the bins.</p></dd>


<dt>density_adjust</dt>
<dd><p>Passed to <code>adjust</code> argument in 
<a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a>.</p></dd>


<dt>zoom</dt>
<dd><p>Zoom into the center of the locus when plotting
(without editing the fine-mapping results file).
You can provide either:</p><ul><li><p>The size of your plot window in terms of basepairs
(e.g. <code>zoom=50000</code> for a 50kb window).</p></li>
<li><p>How much you want to zoom in (e.g. <code>zoom="1x"</code>
for the full locus, <code>zoom="2x"</code>
for 2x zoom into the center of the locus, etc.).</p></li>
</ul><p>You can pass a list of window sizes (e.g. <code>c(50000,100000,500000)</code>)
to automatically generate
multiple views of each locus.
This can even be a mix of different style inputs: e.g.
 <code>c("1x","4.5x",25000)</code>.</p></dd>


<dt>strip.text.y.angle</dt>
<dd><p>Angle of the y-axis facet labels.</p></dd>


<dt>xtext</dt>
<dd><p>Whether to include x-axis title and text.</p></dd>


<dt>geom</dt>
<dd><p>Geom to use (Single character for now). Please see section Geometry
    for details.</p></dd>


<dt>plot_formula</dt>
<dd><p>Formula passed to <code>facets</code> argument in 
<a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a>.</p></dd>


<dt>fill_var</dt>
<dd><p>Variable name to use for plot <code>fill</code> argument.</p></dd>


<dt>genomic_units</dt>
<dd><p>Which genomic units to return window limits in.</p></dd>


<dt>as_ggplot</dt>
<dd><p>Return plot as <code>ggplot2</code> 
(<code>TRUE</code>) or <code>Tracks</code> (<code>FALSE</code>) object.</p></dd>


<dt>dpi</dt>
<dd><p>dpi to use for raster graphics</p></dd>


<dt>height</dt>
<dd><p>height (defaults to the height of current
  plotting window)</p></dd>


<dt>width</dt>
<dd><p>width (defaults to the width of current
  plotting window)</p></dd>


<dt>nThread</dt>
<dd><p>Number of threads to parallelise downloads across.</p></dd>


<dt>save_annot</dt>
<dd><p>Save the queried subset of bigwig annotations.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other NOTT2019: 
<code><a href="NOTT2019_bigwig_metadata.html">NOTT2019_bigwig_metadata</a></code>,
<code><a href="NOTT2019_get_epigenomic_peaks.html">NOTT2019_get_epigenomic_peaks</a>()</code>,
<code><a href="NOTT2019_get_interactions.html">NOTT2019_get_interactions</a>()</code>,
<code><a href="NOTT2019_get_interactome.html">NOTT2019_get_interactome</a>()</code>,
<code><a href="NOTT2019_get_promoter_celltypes.html">NOTT2019_get_promoter_celltypes</a>()</code>,
<code><a href="NOTT2019_get_promoter_interactome_data.html">NOTT2019_get_promoter_interactome_data</a>()</code>,
<code><a href="NOTT2019_get_regulatory_regions.html">NOTT2019_get_regulatory_regions</a>()</code>,
<code><a href="NOTT2019_plac_seq_plot.html">NOTT2019_plac_seq_plot</a>()</code>,
<code><a href="NOTT2019_superenhancers.html">NOTT2019_superenhancers</a>()</code>,
<code><a href="get_NOTT2019_interactome.html">get_NOTT2019_interactome</a>()</code>,
<code><a href="get_NOTT2019_superenhancer_interactome.html">get_NOTT2019_superenhancer_interactome</a>()</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">nott2019_track</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu">NOTT2019_epigenomic_histograms</span><span class="op">(</span></span></span>
<span class="r-in"><span>    dat <span class="op">=</span> <span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1.html" class="external-link">BST1</a></span>, </span></span>
<span class="r-in"><span>    bigwig_metadata <span class="op">=</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="va"><a href="NOTT2019_bigwig_metadata.html">NOTT2019_bigwig_metadata</a></span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="op">]</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> NOTT2019:: Creating epigenomic histograms plot</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Inferring genomic limits for window: 1x</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing GRanges query using min/max ranges across one or more chromosomes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Downloading data from UCSC.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing... [1] exvivo_H3K27ac_tbp</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing previously downloaded files: /github/home/.cache/R/echoannot/NOTT2019_epigenomic_peaks.rds</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ NOTT2019:: 634,540 ranges retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> dat is already a GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 510 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Calculating max histogram height</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Converting label units to Mb.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using coord:genome to parse x scale</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using coord:genome to parse x scale</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>longer object length is not a multiple of shorter object length</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found more than one class "simpleUnit" in cache; using the first, from namespace 'hexbin'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Also defined by 'ggbio'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found more than one class "unit" in cache; using the first, from namespace 'hexbin'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Also defined by 'ggbio'</span>
<span class="r-plt img"><img src="NOTT2019_epigenomic_histograms-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

