<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot overlap between some SNP group and various epigenomic data — peak_overlap_plot • echoannot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot overlap between some SNP group and various epigenomic data — peak_overlap_plot"><meta property="og:description" content="Plot overlap between some SNP group and various epigenomic data"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot overlap between some SNP group and various epigenomic data</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/R/peak_overlap_plot.R" class="external-link"><code>R/peak_overlap_plot.R</code></a></small>
    <div class="hidden name"><code>peak_overlap_plot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot overlap between some SNP group and various epigenomic data</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">peak_overlap_plot</span><span class="op">(</span></span>
<span>  <span class="va">merged_DT</span>,</span>
<span>  snp_filter <span class="op">=</span> <span class="st">"Consensus_SNP==TRUE"</span>,</span>
<span>  force_new <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  include.NOTT2019_peaks <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include.NOTT2019_enhancers_promoters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include.NOTT2019_PLACseq <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include.CORCES2020_scATACpeaks <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include.CORCES2020_Cicero_coaccess <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include.CORCES2020_bulkATACpeaks <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include.CORCES2020_HiChIP_FitHiChIP_coaccess <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include.CORCES2020_gene_annotations <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  plot_celltype_specificity <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  plot_celltype_specificity_genes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  facets_formula <span class="op">=</span> <span class="st">". ~ Cell_type"</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  label_yaxis <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  x_strip_angle <span class="op">=</span> <span class="fl">90</span>,</span>
<span>  x_tick_angle <span class="op">=</span> <span class="fl">40</span>,</span>
<span>  drop_empty_cols <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  fill_title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">snp_filter</span>, <span class="st">"\nin epigenomic peaks"</span><span class="op">)</span>,</span>
<span>  save_path <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  height <span class="op">=</span> <span class="fl">11</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  subplot_widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://doi.org/10.1126/science.aay0793" class="external-link">
Nott et al., 2019 (The Lancet Neurology)</a>
<a href="https://doi.org/10.1038/s41588-020-00721-x" class="external-link">
Corces et al., 2020 (Nature Genetics)</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>force_new</dt>
<dd><p>Don't use previously downloaded files.</p></dd>


<dt>include.NOTT2019_peaks</dt>
<dd><p>Plot SNP subset overlap with
peaks from cell-type-specific bulk ATAC, H3K27ac, and H3K4me3 assays.</p></dd>


<dt>include.NOTT2019_enhancers_promoters</dt>
<dd><p>Plot SNP subset overlap with
cell enhancers and promoters.</p></dd>


<dt>include.CORCES2020_scATACpeaks</dt>
<dd><p>Plot SNP subset overlap with
cell-type-specific scATAC-seq peaks.</p></dd>


<dt>include.CORCES2020_Cicero_coaccess</dt>
<dd><p>Plot SNP subset overlap with
Cicero coaccessibility peaks (derived from scATACseq).</p></dd>

</dl></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other summarise: 
<code><a href="CS_bin_plot.html">CS_bin_plot</a>()</code>,
<code><a href="CS_counts_plot.html">CS_counts_plot</a>()</code>,
<code><a href="plot_dataset_overlap.html">plot_dataset_overlap</a>()</code>,
<code><a href="super_summary_plot.html">super_summary_plot</a>()</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Data ####</span></span></span>
<span class="r-in"><span><span class="va">merged_DT</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/echodata/man/get_Nalls2019_merged.html" class="external-link">get_Nalls2019_merged</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#### Consensus SNPs #####</span></span></span>
<span class="r-in"><span><span class="va">gg_peaks</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu">peak_overlap_plot</span><span class="op">(</span></span></span>
<span class="r-in"><span>    merged_DT <span class="op">=</span> <span class="va">merged_DT</span>,</span></span>
<span class="r-in"><span>    plot_celltype_specificity <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing previously downloaded files: /github/home/.cache/R/echoannot/NOTT2019_epigenomic_peaks.rds</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ NOTT2019:: 634,540 ranges retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 113 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ NOTT2019:: Getting regulatory regions data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Astrocyte enhancers ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Astrocyte promoters ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Neuronal enhancers ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Neuronal promoters ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Oligo enhancers ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Oligo promoters ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Microglia enhancers ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Microglia promoters ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 48 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ NOTT2019:: Getting interaction anchors data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Microglia interactome ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Neuronal interactome ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing Oligo interactome ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 52 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 44 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CORCES2020:: Extracting overlapping cell-type-specific scATAC-seq peaks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 13 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + CORCES2020:: Cicero coaccessibility scores identified for 2 / 13 peak hits.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CORCES2020:: Annotating peaks by cell-type-specific target genes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CORCES2020:: Extracting overlapping bulkATAC-seq peaks from brain tissue</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> CORCES2020:: Annotating peaks by bulk brain target genes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 70 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 72 query SNP(s) detected with reference overlap.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + CORCES2020:: Found 142 hits with HiChIP_FitHiChIP coaccessibility loop anchors.</span>
<span class="r-plt img"><img src="peak_overlap_plot-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co">#### UCS SNPs ####</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">gg_peaks</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu">peak_overlap_plot</span><span class="op">(</span></span></span>
<span class="r-in"><span>    merged_DT <span class="op">=</span> <span class="va">merged_DT</span>,</span></span>
<span class="r-in"><span>    snp_filter <span class="op">=</span> <span class="st">"Support&gt;0"</span>,</span></span>
<span class="r-in"><span>    fill_title <span class="op">=</span> <span class="st">"UCS SNPs in epigenomic peaks"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

