<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ echoannot</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echoannot/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="CORCES_2020.HiChIP_FitHiChIP_loop_calls.html">CORCES_2020.HiChIP_FitHiChIP_loop_calls</a></code> </p>
        </td>
        <td><p>FitHiChIP loop calls from human brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.bulkATACseq_peaks.html">CORCES_2020.bulkATACseq_peaks</a></code> </p>
        </td>
        <td><p>bulkATACseq peaks from human brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.cicero_coaccessibility.html">CORCES_2020.cicero_coaccessibility</a></code> </p>
        </td>
        <td><p>Cicero_coaccessibility from human brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.get_ATAC_peak_overlap.html">CORCES_2020.get_ATAC_peak_overlap()</a></code> </p>
        </td>
        <td><p>Get overlap between datatable of SNPs and scATAC peaks</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.get_HiChIP_FitHiChIP_overlap.html">CORCES_2020.get_HiChIP_FitHiChIP_overlap()</a></code> </p>
        </td>
        <td><p>Get overlap between data table of SNPs and HiChIP_FitHiChIP
coaccessibility anchors</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.scATACseq_celltype_peaks.html">CORCES_2020.scATACseq_celltype_peaks</a></code> </p>
        </td>
        <td><p>scATACseq cell type-specific peaks from human brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CORCES_2020.scATACseq_peaks.html">CORCES_2020.scATACseq_peaks</a></code> </p>
        </td>
        <td><p>scATACseq peaks from human brain tissue</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CS_bin_plot.html">CS_bin_plot()</a></code> </p>
        </td>
        <td><p>Plot CS bin counts</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CS_counts_plot.html">CS_counts_plot()</a></code> </p>
        </td>
        <td><p>Bar plot of tool-specific CS sizes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="NOTT_2019.bigwig_metadata.html">NOTT_2019.bigwig_metadata</a></code> </p>
        </td>
        <td><p>Metadata and links to data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="NOTT_2019.interactome.html">NOTT_2019.interactome</a></code> </p>
        </td>
        <td><p>Brain cell type-specific enhancers, promoters, and interactomes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="NOTT_2019.superenhancer_interactome.html">NOTT_2019.superenhancer_interactome</a></code> </p>
        </td>
        <td><p>Brain cell type-specific interactomes with superenhancers</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.construct_reference.html">ROADMAP.construct_reference()</a></code> </p>
        </td>
        <td><p>Gather Roadmap annotation metadata</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.merge_and_process_grl.html">ROADMAP.merge_and_process_grl()</a></code> </p>
        </td>
        <td><p>Standardize Roadmap query</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.query.html">ROADMAP.query()</a></code> </p>
        </td>
        <td><p>Query Roadmap by genomic coordinates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ROADMAP.query_and_plot.html">ROADMAP.query_and_plot()</a></code> </p>
        </td>
        <td><p>Query and plot Roadmap epigenomic annotations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="XGR.download_and_standardize.html">XGR.download_and_standardize()</a></code> </p>
        </td>
        <td><p>Download, standardize, and merge XGR annotations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="XGR.enrichment.html">XGR.enrichment()</a></code> </p>
        </td>
        <td><p>XGR enrichment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="XGR.enrichment_bootstrap.html">XGR.enrichment_bootstrap()</a></code> </p>
        </td>
        <td><p>XGR enrichment (bootstrapped)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="XGR.enrichment_plot.html">XGR.enrichment_plot()</a></code> </p>
        </td>
        <td><p>Plot enrichment results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="annotate_missense.html">annotate_missense()</a></code> </p>
        </td>
        <td><p>Annotate any missense variants</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_SNPgroup_counts.html">get_SNPgroup_counts()</a></code> </p>
        </td>
        <td><p>Tally locus-specific SNP group sizes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="merge_celltype_specific_epigenomics.html">merge_celltype_specific_epigenomics()</a></code> </p>
        </td>
        <td><p>Merge all cell-type-specific epigenomics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="merge_finemapping_results.html">merge_finemapping_results()</a></code> </p>
        </td>
        <td><p>Merge fine-mapping results from all loci</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_dataset_overlap.html">plot_dataset_overlap()</a></code> </p>
        </td>
        <td><p>Plot inter-study SNP overlap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_missense.html">plot_missense()</a></code> </p>
        </td>
        <td><p>Plot any missense variants</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="super_summary_plot.html">super_summary_plot()</a></code> </p>
        </td>
        <td><p>Merge all summary plots into one super plot</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


