<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot overlap between some SNP group and various epigenomic data — peak_overlap_plot • echoannot</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Plot overlap between some SNP group and various epigenomic data — peak_overlap_plot" />
<meta property="og:description" content="Plot overlap between some SNP group and various epigenomic data" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RajLabMSSM/echoannot/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot overlap between some SNP group and various epigenomic data</h1>
    <small class="dont-index">Source: <a href='https://github.com/RajLabMSSM/echoannot/blob/master/R/peak_overlap_plot.R'><code>R/peak_overlap_plot.R</code></a></small>
    <div class="hidden name"><code>peak_overlap_plot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot overlap between some SNP group and various epigenomic data</p>
    </div>

    <pre class="usage"><span class='fu'>peak_overlap_plot</span><span class='op'>(</span>
  <span class='va'>merged_DT</span>,
  snp_filter <span class='op'>=</span> <span class='st'>"Consensus_SNP==TRUE"</span>,
  include.NOTT_2019_peaks <span class='op'>=</span> <span class='cn'>TRUE</span>,
  include.NOTT_2019_enhancers_promoters <span class='op'>=</span> <span class='cn'>TRUE</span>,
  include.NOTT_2019_PLACseq <span class='op'>=</span> <span class='cn'>TRUE</span>,
  include.CORCES_2020_scATACpeaks <span class='op'>=</span> <span class='cn'>TRUE</span>,
  include.CORCES_2020_Cicero_coaccess <span class='op'>=</span> <span class='cn'>TRUE</span>,
  include.CORCES_2020_bulkATACpeaks <span class='op'>=</span> <span class='cn'>TRUE</span>,
  include.CORCES_2020_HiChIP_FitHiChIP_coaccess <span class='op'>=</span> <span class='cn'>TRUE</span>,
  include.CORCES_2020_gene_annotations <span class='op'>=</span> <span class='cn'>TRUE</span>,
  plot_celltype_specificity <span class='op'>=</span> <span class='cn'>TRUE</span>,
  plot_celltype_specificity_genes <span class='op'>=</span> <span class='cn'>FALSE</span>,
  facets_formula <span class='op'>=</span> <span class='st'>". ~ Cell_type"</span>,
  show_plot <span class='op'>=</span> <span class='cn'>TRUE</span>,
  label_yaxis <span class='op'>=</span> <span class='cn'>TRUE</span>,
  x_strip_angle <span class='op'>=</span> <span class='fl'>90</span>,
  x_tick_angle <span class='op'>=</span> <span class='fl'>40</span>,
  drop_empty_cols <span class='op'>=</span> <span class='cn'>F</span>,
  fill_title <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>snp_filter</span>, <span class='st'>"\nin epigenomic peaks"</span><span class='op'>)</span>,
  save_path <span class='op'>=</span> <span class='cn'>F</span>,
  height <span class='op'>=</span> <span class='fl'>11</span>,
  width <span class='op'>=</span> <span class='fl'>12</span>,
  subplot_widths <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>0.5</span><span class='op'>)</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>include.NOTT_2019_peaks</th>
      <td><p>Plot SNP subset overlap with
peaks from cell-type-specific bulk ATAC, H3K27ac, and H3K4me3 assays.</p></td>
    </tr>
    <tr>
      <th>include.NOTT_2019_enhancers_promoters</th>
      <td><p>Plot SNP subset overlap with
cell enhancers and promoters.</p></td>
    </tr>
    <tr>
      <th>include.CORCES_2020_scATACpeaks</th>
      <td><p>Plot SNP subset overlap with
cell-type-specific scATAC-seq peaks.</p></td>
    </tr>
    <tr>
      <th>include.CORCES_2020_Cicero_coaccess</th>
      <td><p>Plot SNP subset overlap with
Cicero coaccessibility peaks (derived from scATACseq).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="source"><a class="anchor" href="#source"></a>Source</h2>

    <p><a href='https://science.sciencemag.org/content/366/6469/1134'>
Nott et al. (2019)</a>
<a href='https://www.biorxiv.org/content/10.1101/2020.01.06.896159v1'>
Corces et al. (2020/bioRxiv)</a></p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other summarise: 
<code><a href='CS_bin_plot.html'>CS_bin_plot</a>()</code>,
<code><a href='CS_counts_plot.html'>CS_counts_plot</a>()</code>,
<code><a href='get_CS_bins.html'>get_CS_bins</a>()</code>,
<code><a href='get_CS_counts.html'>get_CS_counts</a>()</code>,
<code><a href='get_SNPgroup_counts.html'>get_SNPgroup_counts</a>()</code>,
<code><a href='plot_dataset_overlap.html'>plot_dataset_overlap</a>()</code>,
<code><a href='results_report.html'>results_report</a>()</code>,
<code><a href='super_summary_plot.html'>super_summary_plot</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Consensus SNPs</span>
<span class='va'>gg_peaks</span> <span class='op'>&lt;-</span> <span class='fu'>peak_overlap_plot</span><span class='op'>(</span>
    merged_DT <span class='op'>=</span> <span class='fu'>echodata</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/echodata/man/Nalls2019_merged.html'>Nalls2019_merged</a></span>,
    snp_filter <span class='op'>=</span> <span class='st'>"Consensus_SNP==T"</span>,
    fill_title <span class='op'>=</span> <span class='st'>"Consensus SNPs in epigenomic peaks"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading and merging 12 peaks BED files.</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading ATAC/NeuN_optimal_peak_IDR_ENCODE.ATAC.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading ATAC/PU1_optimal_peak_IDR_ENCODE.ATAC.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading ATAC/Olig2_optimal_peak_IDR_ENCODE.ATAC.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading ATAC/LHX2_optimal_peak_IDR_ENCODE.ATAC.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K27ac/NeuN_optimal_peak.H3K27.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K27ac/PU1_optimal_peak.H3K27.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K27ac/Olig2_optimal_peak.H3K27.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K27ac/LHX2_optimal_peak.H3K27.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K4me3/NeuN_optimal_peak.H3K4me3.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K4me3/PU1_optimal_peak.H3K4me3.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K4me3/Olig2_optimal_peak.H3K4me3.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K4me3/LHX2_optimal_peak.H3K4me3.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Converting merged BED files to GRanges.</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: 634540 ranges retrieved.</span></div><div class='output co'>#&gt; <span class='message'>140 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: The `.dots` argument of `group_by()` is deprecated as of dplyr 1.0.0.</span></div><div class='output co'>#&gt; <span class='message'>Importing Astrocyte enhancers ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Astrocyte promoters ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Neuronal enhancers ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Neuronal promoters ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Oligo enhancers ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Oligo promoters ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Microglia enhancers ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Microglia promoters ...</span></div><div class='output co'>#&gt; <span class='message'>++ Converting to GRanges.</span></div><div class='output co'>#&gt; <span class='message'>108 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>Importing Microglia interactome ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Neuronal interactome ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Oligo interactome ...</span></div><div class='output co'>#&gt; <span class='message'>210 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>169 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>CORCES_2020:: Extracting overlapping cell-type-specific scATAC-seq peaks</span></div><div class='output co'>#&gt; <span class='message'>Performing liftover: hg38 ==&gt; hg19</span></div><div class='output co'>#&gt; <span class='message'>/var/folders/zq/h7mtybc533b1qzkys_ttgpth0000gn/T//RtmprlbVSo/hg19ToHg38.over.chain.gz</span></div><div class='output co'>#&gt; <span class='message'>4 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>+ CORCES_2020:: No Cicero hits found.</span></div><div class='output co'>#&gt; <span class='message'>CORCES_2020:: Annotating peaks by cell-type-specific target genes</span></div><div class='output co'>#&gt; <span class='message'>CORCES_2020:: Extracting overlapping bulkATAC-seq peaks from brain tissue</span></div><div class='output co'>#&gt; <span class='message'>Performing liftover: hg38 ==&gt; hg19</span></div><div class='output co'>#&gt; <span class='message'>3 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>CORCES_2020:: Annotating peaks by bulk brain target genes</span></div><div class='output co'>#&gt; Error in if (!is.null(convert_ref_genome) &amp;&amp; toupper(convert_ref_genome) !=  : 
#&gt;   missing value where TRUE/FALSE needed</div><div class='img'><img src='peak_overlap_plot-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'># UCS SNPs</span>
<span class='va'>gg_peaks</span> <span class='op'>&lt;-</span> <span class='fu'>peak_overlap_plot</span><span class='op'>(</span>
    merged_DT <span class='op'>=</span> <span class='fu'>echodata</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/echodata/man/Nalls2019_merged.html'>Nalls2019_merged</a></span>,
    snp_filter <span class='op'>=</span> <span class='st'>"Support&gt;0"</span>,
    fill_title <span class='op'>=</span> <span class='st'>"UCS SNPs in epigenomic peaks"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading and merging 12 peaks BED files.</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading ATAC/NeuN_optimal_peak_IDR_ENCODE.ATAC.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading ATAC/PU1_optimal_peak_IDR_ENCODE.ATAC.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading ATAC/Olig2_optimal_peak_IDR_ENCODE.ATAC.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading ATAC/LHX2_optimal_peak_IDR_ENCODE.ATAC.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K27ac/NeuN_optimal_peak.H3K27.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K27ac/PU1_optimal_peak.H3K27.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K27ac/Olig2_optimal_peak.H3K27.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K27ac/LHX2_optimal_peak.H3K27.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K4me3/NeuN_optimal_peak.H3K4me3.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K4me3/PU1_optimal_peak.H3K4me3.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K4me3/Olig2_optimal_peak.H3K4me3.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Downloading H3K4me3/LHX2_optimal_peak.H3K4me3.bed</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: Converting merged BED files to GRanges.</span></div><div class='output co'>#&gt; <span class='message'>++ NOTT_2019:: 634540 ranges retrieved.</span></div><div class='output co'>#&gt; <span class='message'>386 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>Importing Astrocyte enhancers ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Astrocyte promoters ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Neuronal enhancers ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Neuronal promoters ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Oligo enhancers ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Oligo promoters ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Microglia enhancers ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Microglia promoters ...</span></div><div class='output co'>#&gt; <span class='message'>++ Converting to GRanges.</span></div><div class='output co'>#&gt; <span class='message'>311 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>Importing Microglia interactome ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Neuronal interactome ...</span></div><div class='output co'>#&gt; <span class='message'>Importing Oligo interactome ...</span></div><div class='output co'>#&gt; <span class='message'>861 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>699 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>CORCES_2020:: Extracting overlapping cell-type-specific scATAC-seq peaks</span></div><div class='output co'>#&gt; <span class='message'>Performing liftover: hg38 ==&gt; hg19</span></div><div class='output co'>#&gt; <span class='message'>16 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>+ CORCES_2020:: Cicero coaccessibility scores identified for 1 / 16 peak hits.</span></div><div class='output co'>#&gt; <span class='message'>CORCES_2020:: Annotating peaks by cell-type-specific target genes</span></div><div class='output co'>#&gt; <span class='message'>CORCES_2020:: Extracting overlapping bulkATAC-seq peaks from brain tissue</span></div><div class='output co'>#&gt; <span class='message'>Performing liftover: hg38 ==&gt; hg19</span></div><div class='output co'>#&gt; <span class='message'>12 query SNP(s) detected with reference overlap.</span></div><div class='output co'>#&gt; <span class='message'>CORCES_2020:: Annotating peaks by bulk brain target genes</span></div><div class='output co'>#&gt; Error in if (!is.null(convert_ref_genome) &amp;&amp; toupper(convert_ref_genome) !=  : 
#&gt;   missing value where TRUE/FALSE needed</div><div class='img'><img src='peak_overlap_plot-2.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


