<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Cell-type-specific epigenomics • echoannot</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Cell-type-specific epigenomics">
<meta property="og:description" content="echoannot">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Cell-type-specific epigenomics</h1>
                        <h4 data-toc-skip class="author"><h4> ¶ Author:
<i>Brian M. Schilder</i> ¶ </h4></h4>
            
            <h4 data-toc-skip class="date"><h4> ¶ Updated:
<i>Oct-30-2022</i> ¶ </h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/vignettes/cell_type_specific_epigenomics.Rmd" class="external-link"><code>vignettes/cell_type_specific_epigenomics.Rmd</code></a></small>
      <div class="hidden name"><code>cell_type_specific_epigenomics.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echoannot" class="external-link">echoannot</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="nott2019">Nott2019<a class="anchor" aria-label="anchor" href="#nott2019"></a>
</h3>
<p><code>echoannot</code> includes data generated by <a href="https://doi.org/10.1126/science.aay0793" class="external-link">“Nott2019”</a>:</p>
<blockquote>
<p>Nott A, Holtman IR, Coufal NG, … Glass CK. Brain cell type-specific
enhancer-promoter interactome maps and disease-risk association.
Science. 2019 Nov 29;366(6469):1134-1139. doi: 10.1126/science.aay0793.
Epub 2019 Nov 14. PMID: 31727856; PMCID: PMC7028213.</p>
</blockquote>
<div class="section level4">
<h4 id="import-data">Import data<a class="anchor" aria-label="anchor" href="#import-data"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">superenhancers</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/get_NOTT2019_superenhancer_interactome.html">get_NOTT2019_superenhancer_interactome</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">enhancers_promoters</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/NOTT2019_get_promoter_interactome_data.html">NOTT2019_get_promoter_interactome_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/NOTT2019_get_epigenomic_peaks.html">NOTT2019_get_epigenomic_peaks</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
</div>
<div class="section level4">
<h4 id="plot">Plot<a class="anchor" aria-label="anchor" href="#plot"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1.html" class="external-link">BST1</a></span></span>
<span><span class="va">histo_out</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/NOTT2019_epigenomic_histograms.html">NOTT2019_epigenomic_histograms</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## NOTT2019:: Creating epigenomic histograms plot</span></span></code></pre>
<pre><code><span><span class="co">## + Inferring genomic limits for window: 1x</span></span></code></pre>
<pre><code><span><span class="co">## Constructing GRanges query using min/max ranges across one or more chromosomes.</span></span></code></pre>
<pre><code><span><span class="co">## Downloading data from UCSC.</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [1] exvivo_H3K27ac_tbp</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [2] microglia_H3K27ac</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [3] neurons_H3K27ac</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [4] oligodendrocytes_H3K27ac</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [5] astrocytes_H3K27ac</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [6] exvivo_atac_tbp</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [7] microglia_atac</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [8] neurons_atac</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [9] oligodendrocytes_atac</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [10] astrocytes_atac</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [11] microglia_H3K4me3</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [12] neurons_H3K4me3</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [13] oligodendrocytes_H3K4me3</span></span></code></pre>
<pre><code><span><span class="co">## Importing... [14] astrocytes_H3K4me3</span></span></code></pre>
<pre><code><span><span class="co">## Importing previously downloaded files: /github/home/.cache/R/echoannot/NOTT2019_epigenomic_peaks.rds</span></span></code></pre>
<pre><code><span><span class="co">## ++ NOTT2019:: 634,540 ranges retrieved.</span></span></code></pre>
<pre><code><span><span class="co">## dat is already a GRanges object.</span></span></code></pre>
<pre><code><span><span class="co">## 543 query SNP(s) detected with reference overlap.</span></span></code></pre>
<pre><code><span><span class="co">## + Calculating max histogram height</span></span></code></pre>
<pre><code><span><span class="co">## + Converting label units to Mb.</span></span></code></pre>
<pre><code><span><span class="co">## using coord:genome to parse x scale</span></span>
<span><span class="co">## using coord:genome to parse x scale</span></span></code></pre>
<pre><code><span><span class="co">## Warning in !vapply(ggl, fixed, logical(1L)) &amp; !vapply(PlotList, is,</span></span>
<span><span class="co">## "Ideogram", : longer object length is not a multiple of shorter object length</span></span></code></pre>
<pre><code><span><span class="co">## Found more than one class "simpleUnit" in cache; using the first, from namespace 'hexbin'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'ggbio'</span></span></code></pre>
<pre><code><span><span class="co">## Found more than one class "unit" in cache; using the first, from namespace 'hexbin'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'ggbio'</span></span></code></pre>
<p><img src="cell_type_specific_epigenomics_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<p>In addition to the plot object, tables of both raw read ranges and
called peaks are included in the output list.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">histo_out</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">raw</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="10%">
<col width="10%">
<col width="10%">
<col width="7%">
<col width="8%">
<col width="7%">
<col width="12%">
<col width="9%">
<col width="22%">
</colgroup>
<thead><tr class="header">
<th align="left">seqnames</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">width</th>
<th align="left">strand</th>
<th align="right">score</th>
<th align="left">Cell_type</th>
<th align="left">Assay</th>
<th align="left">Experiment</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">chr4</td>
<td align="right">14737349</td>
<td align="right">14737411</td>
<td align="right">63</td>
<td align="left">*</td>
<td align="right">0.6</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
</tr>
<tr class="even">
<td align="left">chr4</td>
<td align="right">14737488</td>
<td align="right">14737562</td>
<td align="right">75</td>
<td align="left">*</td>
<td align="right">0.2</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
</tr>
<tr class="odd">
<td align="left">chr4</td>
<td align="right">14737692</td>
<td align="right">14737766</td>
<td align="right">75</td>
<td align="left">*</td>
<td align="right">0.6</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
</tr>
<tr class="even">
<td align="left">chr4</td>
<td align="right">14737782</td>
<td align="right">14737856</td>
<td align="right">75</td>
<td align="left">*</td>
<td align="right">0.4</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
</tr>
<tr class="odd">
<td align="left">chr4</td>
<td align="right">14738054</td>
<td align="right">14738126</td>
<td align="right">73</td>
<td align="left">*</td>
<td align="right">0.6</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
</tr>
<tr class="even">
<td align="left">chr4</td>
<td align="right">14738127</td>
<td align="right">14738128</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="right">1.2</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">histo_out</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">peaks</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="8%">
<col width="8%">
<col width="8%">
<col width="5%">
<col width="6%">
<col width="5%">
<col width="6%">
<col width="9%">
<col width="10%">
<col width="7%">
<col width="16%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left">seqnames</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">width</th>
<th align="left">strand</th>
<th align="left">Assay</th>
<th align="left">Marker</th>
<th align="left">Cell_type</th>
<th align="left">Cell_type.1</th>
<th align="left">Assay.1</th>
<th align="left">Experiment</th>
<th align="right">y</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">4</td>
<td align="right">14745668</td>
<td align="right">14746002</td>
<td align="right">335</td>
<td align="left">*</td>
<td align="left">peaks</td>
<td align="left">Olig2</td>
<td align="left">oligo</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
<td align="right">-1.1e-06</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">14751439</td>
<td align="right">14751837</td>
<td align="right">399</td>
<td align="left">*</td>
<td align="left">peaks</td>
<td align="left">Olig2</td>
<td align="left">oligo</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
<td align="right">-1.1e-06</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="right">14768551</td>
<td align="right">14768735</td>
<td align="right">185</td>
<td align="left">*</td>
<td align="left">peaks</td>
<td align="left">PU1</td>
<td align="left">microglia</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
<td align="right">-1.1e-06</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">14768704</td>
<td align="right">14769257</td>
<td align="right">554</td>
<td align="left">*</td>
<td align="left">peaks</td>
<td align="left">PU1</td>
<td align="left">microglia</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
<td align="right">-1.1e-06</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="right">14771450</td>
<td align="right">14773099</td>
<td align="right">1650</td>
<td align="left">*</td>
<td align="left">peaks</td>
<td align="left">LHX2</td>
<td align="left">astrocytes</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
<td align="right">-1.1e-06</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">14829018</td>
<td align="right">14829146</td>
<td align="right">129</td>
<td align="left">*</td>
<td align="left">peaks</td>
<td align="left">NeuN</td>
<td align="left">neurons</td>
<td align="left">microglia</td>
<td align="left">H3K27ac</td>
<td align="left">exvivo H3K27ac tbp</td>
<td align="right">-1.1e-06</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="corces2020">Corces2020<a class="anchor" aria-label="anchor" href="#corces2020"></a>
</h3>
<p><code>echoannot</code> also includes data generated by <a href="https://doi.org/10.1038/s41588-020-00721-x" class="external-link">“Corces2019”</a>:</p>
<blockquote>
<p>Corces, M.R., Shcherbina, A., Kundu, S. et al. Single-cell epigenomic
analyses implicate candidate causal variants at inherited risk loci for
Alzheimer’s and Parkinson’s diseases. Nat Genet 52, 1158–1168 (2020). <a href="https://doi.org/10.1038/s41588-020-00721-x" class="external-link uri">https://doi.org/10.1038/s41588-020-00721-x</a></p>
</blockquote>
<div class="section level4">
<h4 id="import-data-1">Import data<a class="anchor" aria-label="anchor" href="#import-data-1"></a>
</h4>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bulkATACseq_peaks</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/get_CORCES2020_bulkATACseq_peaks.html">get_CORCES2020_bulkATACseq_peaks</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">cicero_coaccessibility</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/get_CORCES2020_cicero_coaccessibility.html">get_CORCES2020_cicero_coaccessibility</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">hichip_fithichip_loop_calls</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/get_CORCES2020_hichip_fithichip_loop_calls.html">get_CORCES2020_hichip_fithichip_loop_calls</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">scATACseq_celltype_peaks</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/get_CORCES2020_scATACseq_celltype_peaks.html">get_CORCES2020_scATACseq_celltype_peaks</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">scATACseq_peaks</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/get_CORCES2020_scATACseq_peaks.html">get_CORCES2020_scATACseq_peaks</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="plot-1">Plot<a class="anchor" aria-label="anchor" href="#plot-1"></a>
</h4>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peak_dat</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/granges_overlap.html">granges_overlap</a></span><span class="op">(</span></span>
<span>    dat1 <span class="op">=</span> <span class="va">dat</span>,  </span>
<span>    chrom_col.1 <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>    start_col.1 <span class="op">=</span> <span class="st">"POS"</span>,</span>
<span>    dat2 <span class="op">=</span> <span class="va">scATACseq_celltype_peaks</span>, </span>
<span>    chrom_col.2 <span class="op">=</span> <span class="st">"hg38_Chromosome"</span>,</span>
<span>    start_col.2 <span class="op">=</span> <span class="st">"hg38_Start"</span>,</span>
<span>    end_col.2 <span class="op">=</span> <span class="st">"hg38_Stop"</span><span class="op">)</span></span>
<span><span class="fu">ggbio</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">peak_dat</span>, </span>
<span>                <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">ExcitatoryNeurons</span>, color<span class="op">=</span><span class="va">Effect</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] echoannot_0.99.10 BiocStyle_2.25.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] utf8_1.2.2                  reticulate_1.26            </span></span>
<span><span class="co">##   [3] R.utils_2.12.0              tidyselect_1.2.0           </span></span>
<span><span class="co">##   [5] RSQLite_2.2.18              AnnotationDbi_1.59.1       </span></span>
<span><span class="co">##   [7] htmlwidgets_1.5.4           grid_4.2.1                 </span></span>
<span><span class="co">##   [9] BiocParallel_1.31.15        XGR_1.1.8                  </span></span>
<span><span class="co">##  [11] munsell_0.5.0               codetools_0.2-18           </span></span>
<span><span class="co">##  [13] ragg_1.2.4                  interp_1.1-3               </span></span>
<span><span class="co">##  [15] DT_0.26                     colorspace_2.0-3           </span></span>
<span><span class="co">##  [17] OrganismDbi_1.39.1          Biobase_2.57.3             </span></span>
<span><span class="co">##  [19] filelock_1.0.2              highr_0.9                  </span></span>
<span><span class="co">##  [21] knitr_1.40                  supraHex_1.35.0            </span></span>
<span><span class="co">##  [23] rstudioapi_0.14             stats4_4.2.1               </span></span>
<span><span class="co">##  [25] DescTools_0.99.47           labeling_0.4.2             </span></span>
<span><span class="co">##  [27] MatrixGenerics_1.9.1        GenomeInfoDbData_1.2.9     </span></span>
<span><span class="co">##  [29] farver_2.1.1                bit64_4.0.5                </span></span>
<span><span class="co">##  [31] echoconda_0.99.7            rprojroot_2.0.3            </span></span>
<span><span class="co">##  [33] basilisk_1.9.12             vctrs_0.5.0                </span></span>
<span><span class="co">##  [35] generics_0.1.3              xfun_0.34                  </span></span>
<span><span class="co">##  [37] biovizBase_1.45.0           BiocFileCache_2.5.2        </span></span>
<span><span class="co">##  [39] R6_2.5.1                    GenomeInfoDb_1.33.16       </span></span>
<span><span class="co">##  [41] AnnotationFilter_1.21.0     bitops_1.0-7               </span></span>
<span><span class="co">##  [43] cachem_1.0.6                reshape_0.8.9              </span></span>
<span><span class="co">##  [45] DelayedArray_0.23.2         assertthat_0.2.1           </span></span>
<span><span class="co">##  [47] BiocIO_1.7.1                scales_1.2.1               </span></span>
<span><span class="co">##  [49] nnet_7.3-18                 rootSolve_1.8.2.3          </span></span>
<span><span class="co">##  [51] gtable_0.3.1                lmom_2.9                   </span></span>
<span><span class="co">##  [53] ggbio_1.45.0                ensembldb_2.21.5           </span></span>
<span><span class="co">##  [55] rlang_1.0.6                 systemfonts_1.0.4          </span></span>
<span><span class="co">##  [57] echodata_0.99.15            splines_4.2.1              </span></span>
<span><span class="co">##  [59] lazyeval_0.2.2              rtracklayer_1.57.0         </span></span>
<span><span class="co">##  [61] dichromat_2.0-0.1           hexbin_1.28.2              </span></span>
<span><span class="co">##  [63] checkmate_2.1.0             BiocManager_1.30.19        </span></span>
<span><span class="co">##  [65] yaml_2.3.6                  reshape2_1.4.4             </span></span>
<span><span class="co">##  [67] backports_1.4.1             GenomicFeatures_1.49.8     </span></span>
<span><span class="co">##  [69] ggnetwork_0.5.10            Hmisc_4.7-1                </span></span>
<span><span class="co">##  [71] RBGL_1.73.0                 tools_4.2.1                </span></span>
<span><span class="co">##  [73] bookdown_0.29               ggplot2_3.3.6              </span></span>
<span><span class="co">##  [75] ellipsis_0.3.2              jquerylib_0.1.4            </span></span>
<span><span class="co">##  [77] RColorBrewer_1.1-3          proxy_0.4-27               </span></span>
<span><span class="co">##  [79] BiocGenerics_0.43.4         Rcpp_1.0.9                 </span></span>
<span><span class="co">##  [81] plyr_1.8.7                  base64enc_0.1-3            </span></span>
<span><span class="co">##  [83] progress_1.2.2              zlibbioc_1.43.0            </span></span>
<span><span class="co">##  [85] purrr_0.3.5                 RCurl_1.98-1.9             </span></span>
<span><span class="co">##  [87] basilisk.utils_1.9.4        prettyunits_1.1.1          </span></span>
<span><span class="co">##  [89] rpart_4.1.16                deldir_1.0-6               </span></span>
<span><span class="co">##  [91] S4Vectors_0.35.4            cluster_2.1.4              </span></span>
<span><span class="co">##  [93] SummarizedExperiment_1.27.3 ggrepel_0.9.1              </span></span>
<span><span class="co">##  [95] fs_1.5.2                    crul_1.3                   </span></span>
<span><span class="co">##  [97] magrittr_2.0.3              data.table_1.14.4          </span></span>
<span><span class="co">##  [99] echotabix_0.99.8            dnet_1.1.7                 </span></span>
<span><span class="co">## [101] openxlsx_4.2.5.1            mvtnorm_1.1-3              </span></span>
<span><span class="co">## [103] ProtGenerics_1.29.1         matrixStats_0.62.0         </span></span>
<span><span class="co">## [105] patchwork_1.1.2             hms_1.1.2                  </span></span>
<span><span class="co">## [107] evaluate_0.17               XML_3.99-0.11              </span></span>
<span><span class="co">## [109] jpeg_0.1-9                  readxl_1.4.1               </span></span>
<span><span class="co">## [111] IRanges_2.31.2              gridExtra_2.3              </span></span>
<span><span class="co">## [113] compiler_4.2.1              biomaRt_2.53.3             </span></span>
<span><span class="co">## [115] tibble_3.1.8                crayon_1.5.2               </span></span>
<span><span class="co">## [117] R.oo_1.25.0                 htmltools_0.5.3            </span></span>
<span><span class="co">## [119] tzdb_0.3.0                  Formula_1.2-4              </span></span>
<span><span class="co">## [121] tidyr_1.2.1                 expm_0.999-6               </span></span>
<span><span class="co">## [123] Exact_3.2                   DBI_1.1.3                  </span></span>
<span><span class="co">## [125] dbplyr_2.2.1                MASS_7.3-58.1              </span></span>
<span><span class="co">## [127] rappdirs_0.3.3              boot_1.3-28                </span></span>
<span><span class="co">## [129] Matrix_1.5-1                readr_2.1.3                </span></span>
<span><span class="co">## [131] piggyback_0.1.4             cli_3.4.1                  </span></span>
<span><span class="co">## [133] R.methodsS3_1.8.2           parallel_4.2.1             </span></span>
<span><span class="co">## [135] igraph_1.3.5                GenomicRanges_1.49.1       </span></span>
<span><span class="co">## [137] pkgconfig_2.0.3             pkgdown_2.0.6.9000         </span></span>
<span><span class="co">## [139] GenomicAlignments_1.33.1    dir.expiry_1.5.1           </span></span>
<span><span class="co">## [141] RCircos_1.2.2               foreign_0.8-83             </span></span>
<span><span class="co">## [143] osfr_0.2.9                  xml2_1.3.3                 </span></span>
<span><span class="co">## [145] bslib_0.4.0                 XVector_0.37.1             </span></span>
<span><span class="co">## [147] stringr_1.4.1               VariantAnnotation_1.43.3   </span></span>
<span><span class="co">## [149] digest_0.6.30               graph_1.75.1               </span></span>
<span><span class="co">## [151] httpcode_0.3.0              Biostrings_2.65.6          </span></span>
<span><span class="co">## [153] rmarkdown_2.17              cellranger_1.1.0           </span></span>
<span><span class="co">## [155] htmlTable_2.4.1             gld_2.6.6                  </span></span>
<span><span class="co">## [157] restfulr_0.0.15             curl_4.3.3                 </span></span>
<span><span class="co">## [159] Rsamtools_2.13.4            rjson_0.2.21               </span></span>
<span><span class="co">## [161] lifecycle_1.0.3             nlme_3.1-160               </span></span>
<span><span class="co">## [163] jsonlite_1.8.3              viridisLite_0.4.1          </span></span>
<span><span class="co">## [165] desc_1.4.2                  BSgenome_1.65.4            </span></span>
<span><span class="co">## [167] fansi_1.0.3                 downloadR_0.99.5           </span></span>
<span><span class="co">## [169] pillar_1.8.1                lattice_0.20-45            </span></span>
<span><span class="co">## [171] GGally_2.1.2                KEGGREST_1.37.3            </span></span>
<span><span class="co">## [173] fastmap_1.1.0               httr_1.4.4                 </span></span>
<span><span class="co">## [175] survival_3.4-0              glue_1.6.2                 </span></span>
<span><span class="co">## [177] zip_2.2.2                   png_0.1-7                  </span></span>
<span><span class="co">## [179] bit_4.0.4                   Rgraphviz_2.41.2           </span></span>
<span><span class="co">## [181] class_7.3-20                stringi_1.7.8              </span></span>
<span><span class="co">## [183] sass_0.4.2                  blob_1.2.3                 </span></span>
<span><span class="co">## [185] textshaping_0.3.6           latticeExtra_0.6-30        </span></span>
<span><span class="co">## [187] memoise_2.0.1               dplyr_1.0.10               </span></span>
<span><span class="co">## [189] e1071_1.7-12                ape_5.6-2</span></span></code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
