<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started • echoannot</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started">
<meta property="og:description" content="echoannot">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoannot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/echoannot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cell_type_specific_epigenomics.html">Cell-type-specific epigenomics</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RajLabMSSM/echoannot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started</h1>
                        <h4 data-toc-skip class="author"><h4> ¶ Author:
<i>Brian M. Schilder</i> ¶ </h4></h4>
            
            <h4 data-toc-skip class="date"><h4> ¶ Updated:
<i>Oct-30-2022</i> ¶ </h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoannot/blob/HEAD/vignettes/echoannot.Rmd" class="external-link"><code>vignettes/echoannot.Rmd</code></a></small>
      <div class="hidden name"><code>echoannot.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RajLabMSSM/echoannot" class="external-link">echoannot</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Registered S3 method overwritten by 'GGally':</span></span>
<span><span class="co">##   method from   </span></span>
<span><span class="co">##   +.gg   ggplot2</span></span></code></pre>
<div class="section level2">
<h2 id="import-data">Import data<a class="anchor" aria-label="anchor" href="#import-data"></a>
</h2>
<p>To get the full dataset of all fine-mapped Parkinson’s Disease loci,
you can use the following function:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merged_DT</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/echodata/man/get_Nalls2019_merged.html" class="external-link">get_Nalls2019_merged</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="annotate">Annotate<a class="anchor" aria-label="anchor" href="#annotate"></a>
</h2>
<p>Annotate SNP-wise fine-mapping results.</p>
<p>Here, we’re only annotating a small number of SNPs high-confidence
causal SNPs for demo purposes. The more SNPs you supply to
<code>annotate_snps</code>, the longer it will take to query the
selected databases for each SNP.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#### Only query high-confidence fine-mapping SNPs from one locus ####</span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="va">merged_DT</span><span class="op">[</span><span class="va">Locus</span><span class="op">==</span><span class="st">"LRRK2"</span> <span class="op">&amp;</span> <span class="va">Consensus_SNP</span><span class="op">==</span><span class="cn">TRUE</span>,<span class="op">]</span></span>
<span><span class="co">#### Query annotations ####</span></span>
<span><span class="va">dat_annot</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/annotate_snps.html">annotate_snps</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat</span>,</span>
<span>                                      haploreg_annotation <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                      regulomeDB_annotation <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                      biomart_annotation <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## Loading required namespace: haploR</span></span></code></pre>
<pre><code><span><span class="co">## + Gathering annotation data from HaploReg...</span></span></code></pre>
<pre><code><span><span class="co">## ++ Submitting chunk 1 / 1</span></span></code></pre>
<pre><code><span><span class="co">## + Gathering annotation data from HaploReg...</span></span></code></pre>
<pre><code><span><span class="co">## ++ Submitting chunk 1 / 1</span></span></code></pre>
<pre><code><span><span class="co">## No encoding supplied: defaulting to UTF-8.</span></span>
<span><span class="co">## No encoding supplied: defaulting to UTF-8.</span></span>
<span><span class="co">## No encoding supplied: defaulting to UTF-8.</span></span>
<span><span class="co">## No encoding supplied: defaulting to UTF-8.</span></span>
<span><span class="co">## No encoding supplied: defaulting to UTF-8.</span></span></code></pre>
<pre><code><span><span class="co">## + Gathering annotation data from Biomart...</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">dat_annot</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="5%">
<col width="19%">
<col width="41%">
<col width="0%">
<col width="0%">
<col width="9%">
<col width="0%">
<col width="0%">
<col width="2%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="1%">
</colgroup>
<thead><tr class="header">
<th align="left">SNP</th>
<th align="left">Locus</th>
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="right">P</th>
<th align="right">Effect</th>
<th align="right">StdErr</th>
<th align="right">Freq</th>
<th align="right">MAF</th>
<th align="right">N_cases</th>
<th align="right">N_controls</th>
<th align="right">proportion_cases</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="left">leadSNP</th>
<th align="right">ABF.CS</th>
<th align="right">ABF.PP</th>
<th align="right">SUSIE.CS</th>
<th align="right">SUSIE.PP</th>
<th align="right">POLYFUN_SUSIE.CS</th>
<th align="right">POLYFUN_SUSIE.PP</th>
<th align="right">FINEMAP.CS</th>
<th align="right">FINEMAP.PP</th>
<th align="right">Mb</th>
<th align="right">Support</th>
<th align="left">Consensus_SNP</th>
<th align="right">mean.PP</th>
<th align="right">mean.CS</th>
<th align="right">N</th>
<th align="right">t_stat</th>
<th align="right">chr</th>
<th align="left">pos_hg38</th>
<th align="right">r2</th>
<th align="right">D’</th>
<th align="right">is_query_snp</th>
<th align="left">ref</th>
<th align="left">alt</th>
<th align="right">AFR</th>
<th align="right">AMR</th>
<th align="right">ASN</th>
<th align="right">EUR</th>
<th align="left">GERP_cons</th>
<th align="left">SiPhy_cons</th>
<th align="left">Chromatin_States</th>
<th align="left">Chromatin_States_Imputed</th>
<th align="left">Chromatin_Marks</th>
<th align="left">DNAse</th>
<th align="left">Proteins</th>
<th align="left">eQTL</th>
<th align="left">gwas</th>
<th align="left">grasp</th>
<th align="left">Motifs</th>
<th align="left">GENCODE_id</th>
<th align="left">GENCODE_name</th>
<th align="left">GENCODE_direction</th>
<th align="left">GENCODE_distance</th>
<th align="left">RefSeq_id</th>
<th align="left">RefSeq_name</th>
<th align="left">RefSeq_direction</th>
<th align="left">RefSeq_distance</th>
<th align="left">dbSNP_functional_annotation</th>
<th align="left">query_snp_rsid</th>
<th align="left">Promoter_histone_marks</th>
<th align="left">Enhancer_histone_marks</th>
<th align="left">regulome.start</th>
<th align="left">regulome.end</th>
<th align="left">features.ChIP</th>
<th align="left">features.DNase</th>
<th align="left">features.Footprint</th>
<th align="left">features.Footprint_matched</th>
<th align="left">features.IC_matched_max</th>
<th align="left">features.IC_max</th>
<th align="left">features.PWM</th>
<th align="left">features.PWM_matched</th>
<th align="left">features.QTL</th>
<th align="left">regulome_score.probability</th>
<th align="left">regulome_score.ranking</th>
<th align="left">allele</th>
<th align="right">chr_name</th>
<th align="right">chrom_start</th>
<th align="right">chrom_end</th>
<th align="right">chrom_strand</th>
<th align="left">ensembl_gene_stable_id</th>
<th align="left">consequence_type_tv</th>
<th align="left">polyphen_prediction</th>
<th align="left">polyphen_score</th>
<th align="left">sift_prediction</th>
<th align="left">sift_score</th>
<th align="left">reg_consequence_types</th>
<th align="left">validated</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rs11175620</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40610864</td>
<td align="right">0</td>
<td align="right">0.1307</td>
<td align="right">0.0142</td>
<td align="right">0.1200</td>
<td align="right">0.1200</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">FALSE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">40.61086</td>
<td align="right">2</td>
<td align="left">TRUE</td>
<td align="right">0.50</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">9.204225</td>
<td align="right">12</td>
<td align="left">40217062</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">G</td>
<td align="left">C</td>
<td align="right">0.11</td>
<td align="right">0.06</td>
<td align="right">0.00</td>
<td align="right">0.11</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E023,7_Enh;E024,7_Enh;E028,7_Enh;E029,7_Enh;E030,2_TssAFlnk;E031,2_TssAFlnk;E032,7_Enh;E035,1_TssA;E052,7_Enh;E080,7_Enh;E114,7_Enh;E117,7_Enh;E119,7_Enh;E124,2_TssAFlnk</td>
<td align="left">E023,16_EnhW1;E025,19_DNase;E028,19_DNase;E029,14_EnhA2;E030,13_EnhA1;E031,14_EnhA2;E032,22_PromP;E033,16_EnhW1;E035,16_EnhW1;E036,14_EnhA2;E037,22_PromP;E038,22_PromP;E039,22_PromP;E040,22_PromP;E041,22_PromP;E042,22_PromP;E043,22_PromP;E044,22_PromP;E045,22_PromP;E046,22_PromP;E048,22_PromP;E050,22_PromP;E051,16_EnhW1;E057,22_PromP;E062,16_EnhW1;E071,22_PromP;E080,22_PromP;E086,19_DNase;E088,22_PromP;E112,17_EnhW2;E113,16_EnhW1;E114,19_DNase;E117,19_DNase;E124,13_EnhA1</td>
<td align="left">E011,H3K4me1_Enh;E018,H3K4me1_Enh;E023,H3K4me1_Enh;E024,H3K4me1_Enh;E025,H3K4me1_Enh;E028,H3K4me1_Enh;E029,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E032,H3K4me1_Enh;E036,H3K4me1_Enh;E049,H3K4me1_Enh;E050,H3K4me1_Enh;E052,H3K4me1_Enh;E056,H3K4me1_Enh;E057,H3K4me1_Enh;E080,H3K4me1_Enh;E086,H3K4me1_Enh;E088,H3K4me1_Enh;E113,H3K4me1_Enh;E114,H3K4me1_Enh;E117,H3K4me1_Enh;E119,H3K4me1_Enh;E124,H3K4me1_Enh;E023,H3K4me3_Pro;E030,H3K4me3_Pro;E031,H3K4me3_Pro;E035,H3K4me3_Pro;E109,H3K4me3_Pro;E124,H3K4me3_Pro;E025,H3K9ac_Pro;E029,H3K27ac_Enh;E032,H3K27ac_Enh;E080,H3K27ac_Enh;E113,H3K27ac_Enh;E124,H3K27ac_Enh</td>
<td align="left">E006;E021;E029;E046;E080;E086;E088;E117;E120;E124</td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,3.29211855956645e-07;GTEx2015_v6,Artery_Tibial,LRRK2,1.08781866153343e-11;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,1.72128347769633e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,5.31856078341246e-10;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,1.00796507632068e-05;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.06318553434241e-05;Westra2013,Whole_Blood,-,1.3808392059010189E-31;Westra2013,Whole_Blood,LRRK2,2.116806823753094E-5</td>
<td align="left">.</td>
<td align="left">.</td>
<td align="left">BCL_disc5;HDAC2_disc3</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">7947</td>
<td align="left">.</td>
<td align="left">rs11175620</td>
<td align="left">BLD</td>
<td align="left">FAT, ESC, BRST, BLD, MUS, ADRL, LNG, CRVX</td>
<td align="left">40610863</td>
<td align="left">40610864</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.60906</td>
<td align="left">4</td>
<td align="left">G/C</td>
<td align="right">12</td>
<td align="right">40610864</td>
<td align="right">40610864</td>
<td align="right">1</td>
<td align="left">ENSG00000225342</td>
<td align="left">non_coding_transcript_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
<td align="left">1000Genomes,Frequency,gnomAD,TOPMed</td>
</tr>
<tr class="even">
<td align="left">rs11175620</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40610864</td>
<td align="right">0</td>
<td align="right">0.1307</td>
<td align="right">0.0142</td>
<td align="right">0.1200</td>
<td align="right">0.1200</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">FALSE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">40.61086</td>
<td align="right">2</td>
<td align="left">TRUE</td>
<td align="right">0.50</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">9.204225</td>
<td align="right">12</td>
<td align="left">40217062</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">G</td>
<td align="left">C</td>
<td align="right">0.11</td>
<td align="right">0.06</td>
<td align="right">0.00</td>
<td align="right">0.11</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E023,7_Enh;E024,7_Enh;E028,7_Enh;E029,7_Enh;E030,2_TssAFlnk;E031,2_TssAFlnk;E032,7_Enh;E035,1_TssA;E052,7_Enh;E080,7_Enh;E114,7_Enh;E117,7_Enh;E119,7_Enh;E124,2_TssAFlnk</td>
<td align="left">E023,16_EnhW1;E025,19_DNase;E028,19_DNase;E029,14_EnhA2;E030,13_EnhA1;E031,14_EnhA2;E032,22_PromP;E033,16_EnhW1;E035,16_EnhW1;E036,14_EnhA2;E037,22_PromP;E038,22_PromP;E039,22_PromP;E040,22_PromP;E041,22_PromP;E042,22_PromP;E043,22_PromP;E044,22_PromP;E045,22_PromP;E046,22_PromP;E048,22_PromP;E050,22_PromP;E051,16_EnhW1;E057,22_PromP;E062,16_EnhW1;E071,22_PromP;E080,22_PromP;E086,19_DNase;E088,22_PromP;E112,17_EnhW2;E113,16_EnhW1;E114,19_DNase;E117,19_DNase;E124,13_EnhA1</td>
<td align="left">E011,H3K4me1_Enh;E018,H3K4me1_Enh;E023,H3K4me1_Enh;E024,H3K4me1_Enh;E025,H3K4me1_Enh;E028,H3K4me1_Enh;E029,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E032,H3K4me1_Enh;E036,H3K4me1_Enh;E049,H3K4me1_Enh;E050,H3K4me1_Enh;E052,H3K4me1_Enh;E056,H3K4me1_Enh;E057,H3K4me1_Enh;E080,H3K4me1_Enh;E086,H3K4me1_Enh;E088,H3K4me1_Enh;E113,H3K4me1_Enh;E114,H3K4me1_Enh;E117,H3K4me1_Enh;E119,H3K4me1_Enh;E124,H3K4me1_Enh;E023,H3K4me3_Pro;E030,H3K4me3_Pro;E031,H3K4me3_Pro;E035,H3K4me3_Pro;E109,H3K4me3_Pro;E124,H3K4me3_Pro;E025,H3K9ac_Pro;E029,H3K27ac_Enh;E032,H3K27ac_Enh;E080,H3K27ac_Enh;E113,H3K27ac_Enh;E124,H3K27ac_Enh</td>
<td align="left">E006;E021;E029;E046;E080;E086;E088;E117;E120;E124</td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,3.29211855956645e-07;GTEx2015_v6,Artery_Tibial,LRRK2,1.08781866153343e-11;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,1.72128347769633e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,5.31856078341246e-10;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,1.00796507632068e-05;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.06318553434241e-05;Westra2013,Whole_Blood,-,1.3808392059010189E-31;Westra2013,Whole_Blood,LRRK2,2.116806823753094E-5</td>
<td align="left">.</td>
<td align="left">.</td>
<td align="left">BCL_disc5;HDAC2_disc3</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">7947</td>
<td align="left">.</td>
<td align="left">rs11175620</td>
<td align="left">BLD</td>
<td align="left">FAT, ESC, BRST, BLD, MUS, ADRL, LNG, CRVX</td>
<td align="left">40610863</td>
<td align="left">40610864</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.60906</td>
<td align="left">4</td>
<td align="left">G/C</td>
<td align="right">12</td>
<td align="right">40610864</td>
<td align="right">40610864</td>
<td align="right">1</td>
<td align="left">ENSG00000225342</td>
<td align="left">intron_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
<td align="left">1000Genomes,Frequency,gnomAD,TOPMed</td>
</tr>
<tr class="odd">
<td align="left">rs11175620</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40610864</td>
<td align="right">0</td>
<td align="right">0.1307</td>
<td align="right">0.0142</td>
<td align="right">0.1200</td>
<td align="right">0.1200</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">C</td>
<td align="left">G</td>
<td align="left">FALSE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">40.61086</td>
<td align="right">2</td>
<td align="left">TRUE</td>
<td align="right">0.50</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">9.204225</td>
<td align="right">12</td>
<td align="left">40217062</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">G</td>
<td align="left">C</td>
<td align="right">0.11</td>
<td align="right">0.06</td>
<td align="right">0.00</td>
<td align="right">0.11</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E023,7_Enh;E024,7_Enh;E028,7_Enh;E029,7_Enh;E030,2_TssAFlnk;E031,2_TssAFlnk;E032,7_Enh;E035,1_TssA;E052,7_Enh;E080,7_Enh;E114,7_Enh;E117,7_Enh;E119,7_Enh;E124,2_TssAFlnk</td>
<td align="left">E023,16_EnhW1;E025,19_DNase;E028,19_DNase;E029,14_EnhA2;E030,13_EnhA1;E031,14_EnhA2;E032,22_PromP;E033,16_EnhW1;E035,16_EnhW1;E036,14_EnhA2;E037,22_PromP;E038,22_PromP;E039,22_PromP;E040,22_PromP;E041,22_PromP;E042,22_PromP;E043,22_PromP;E044,22_PromP;E045,22_PromP;E046,22_PromP;E048,22_PromP;E050,22_PromP;E051,16_EnhW1;E057,22_PromP;E062,16_EnhW1;E071,22_PromP;E080,22_PromP;E086,19_DNase;E088,22_PromP;E112,17_EnhW2;E113,16_EnhW1;E114,19_DNase;E117,19_DNase;E124,13_EnhA1</td>
<td align="left">E011,H3K4me1_Enh;E018,H3K4me1_Enh;E023,H3K4me1_Enh;E024,H3K4me1_Enh;E025,H3K4me1_Enh;E028,H3K4me1_Enh;E029,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E032,H3K4me1_Enh;E036,H3K4me1_Enh;E049,H3K4me1_Enh;E050,H3K4me1_Enh;E052,H3K4me1_Enh;E056,H3K4me1_Enh;E057,H3K4me1_Enh;E080,H3K4me1_Enh;E086,H3K4me1_Enh;E088,H3K4me1_Enh;E113,H3K4me1_Enh;E114,H3K4me1_Enh;E117,H3K4me1_Enh;E119,H3K4me1_Enh;E124,H3K4me1_Enh;E023,H3K4me3_Pro;E030,H3K4me3_Pro;E031,H3K4me3_Pro;E035,H3K4me3_Pro;E109,H3K4me3_Pro;E124,H3K4me3_Pro;E025,H3K9ac_Pro;E029,H3K27ac_Enh;E032,H3K27ac_Enh;E080,H3K27ac_Enh;E113,H3K27ac_Enh;E124,H3K27ac_Enh</td>
<td align="left">E006;E021;E029;E046;E080;E086;E088;E117;E120;E124</td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,3.29211855956645e-07;GTEx2015_v6,Artery_Tibial,LRRK2,1.08781866153343e-11;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,1.72128347769633e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,5.31856078341246e-10;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,1.00796507632068e-05;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.06318553434241e-05;Westra2013,Whole_Blood,-,1.3808392059010189E-31;Westra2013,Whole_Blood,LRRK2,2.116806823753094E-5</td>
<td align="left">.</td>
<td align="left">.</td>
<td align="left">BCL_disc5;HDAC2_disc3</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">7947</td>
<td align="left">.</td>
<td align="left">rs11175620</td>
<td align="left">BLD</td>
<td align="left">FAT, ESC, BRST, BLD, MUS, ADRL, LNG, CRVX</td>
<td align="left">40610863</td>
<td align="left">40610864</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.60906</td>
<td align="left">4</td>
<td align="left">G/C</td>
<td align="right">12</td>
<td align="right">40610864</td>
<td align="right">40610864</td>
<td align="right">1</td>
<td align="left">ENSG00000188906</td>
<td align="left">intron_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
<td align="left">1000Genomes,Frequency,gnomAD,TOPMed</td>
</tr>
<tr class="even">
<td align="left">rs7294619</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40617202</td>
<td align="right">0</td>
<td align="right">-0.1276</td>
<td align="right">0.0140</td>
<td align="right">0.8783</td>
<td align="right">0.1217</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="left">FALSE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">40.61720</td>
<td align="right">3</td>
<td align="left">TRUE</td>
<td align="right">0.75</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">-9.114286</td>
<td align="right">12</td>
<td align="left">40223400</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">0.11</td>
<td align="right">0.06</td>
<td align="right">0.00</td>
<td align="right">0.11</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E023,7_Enh;E025,7_Enh;E029,1_TssA;E030,2_TssAFlnk;E031,2_TssAFlnk;E032,1_TssA;E033,1_TssA;E035,1_TssA;E046,1_TssA;E050,7_Enh;E051,1_TssA;E062,1_TssA;E063,7_Enh;E066,7_Enh;E068,7_Enh;E074,7_Enh;E076,7_Enh;E078,1_TssA;E086,1_TssA;E103,7_Enh;E116,10_TssBiv;E124,1_TssA</td>
<td align="left">E009,23_PromBiv;E023,22_PromP;E025,22_PromP;E029,4_PromD2;E030,3_PromD1;E031,4_PromD2;E032,4_PromD2;E033,4_PromD2;E034,22_PromP;E035,4_PromD2;E036,4_PromD2;E037,22_PromP;E038,4_PromD2;E039,12_TxEnhW;E040,12_TxEnhW;E041,12_TxEnhW;E042,22_PromP;E043,4_PromD2;E044,4_PromD2;E045,12_TxEnhW;E046,4_PromD2;E047,4_PromD2;E048,22_PromP;E050,4_PromD2;E051,4_PromD2;E052,19_DNase;E053,22_PromP;E055,22_PromP;E057,22_PromP;E061,22_PromP;E062,4_PromD2;E063,22_PromP;E066,4_PromD2;E067,22_PromP;E068,22_PromP;E069,22_PromP;E070,22_PromP;E071,12_TxEnhW;E072,22_PromP;E073,22_PromP;E074,19_DNase;E075,22_PromP;E076,4_PromD2;E077,4_PromD2;E078,4_PromD2;E080,22_PromP;E081,22_PromP;E084,22_PromP;E085,22_PromP;E086,22_PromP;E088,22_PromP;E091,22_PromP;E092,22_PromP;E093,22_PromP;E100,22_PromP;E101,4_PromD2;E102,4_PromD2;E103,22_PromP;E106,22_PromP;E108,22_PromP;E109,22_PromP;E110,22_PromP;E111,16_EnhW1;E112,4_PromD2;E113,4_PromD2;E114,22_PromP;E115,23_PromBiv;E116,4_PromD2;E118,22_PromP;E119,22_PromP;E122,22_PromP;E124,3_PromD1</td>
<td align="left">E023,H3K4me1_Enh;E025,H3K4me1_Enh;E026,H3K4me1_Enh;E027,H3K4me1_Enh;E028,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E032,H3K4me1_Enh;E035,H3K4me1_Enh;E036,H3K4me1_Enh;E038,H3K4me1_Enh;E041,H3K4me1_Enh;E043,H3K4me1_Enh;E047,H3K4me1_Enh;E048,H3K4me1_Enh;E049,H3K4me1_Enh;E050,H3K4me1_Enh;E051,H3K4me1_Enh;E052,H3K4me1_Enh;E061,H3K4me1_Enh;E062,H3K4me1_Enh;E063,H3K4me1_Enh;E066,H3K4me1_Enh;E067,H3K4me1_Enh;E068,H3K4me1_Enh;E069,H3K4me1_Enh;E071,H3K4me1_Enh;E072,H3K4me1_Enh;E073,H3K4me1_Enh;E074,H3K4me1_Enh;E076,H3K4me1_Enh;E077,H3K4me1_Enh;E078,H3K4me1_Enh;E080,H3K4me1_Enh;E087,H3K4me1_Enh;E102,H3K4me1_Enh;E103,H3K4me1_Enh;E108,H3K4me1_Enh;E109,H3K4me1_Enh;E111,H3K4me1_Enh;E114,H3K4me1_Enh;E124,H3K4me1_Enh;E129,H3K4me1_Enh;E023,H3K4me3_Pro;E024,H3K4me3_Pro;E025,H3K4me3_Pro;E029,H3K4me3_Pro;E030,H3K4me3_Pro;E031,H3K4me3_Pro;E032,H3K4me3_Pro;E033,H3K4me3_Pro;E035,H3K4me3_Pro;E036,H3K4me3_Pro;E039,H3K4me3_Pro;E041,H3K4me3_Pro;E043,H3K4me3_Pro;E045,H3K4me3_Pro;E046,H3K4me3_Pro;E047,H3K4me3_Pro;E049,H3K4me3_Pro;E051,H3K4me3_Pro;E052,H3K4me3_Pro;E062,H3K4me3_Pro;E063,H3K4me3_Pro;E065,H3K4me3_Pro;E066,H3K4me3_Pro;E067,H3K4me3_Pro;E068,H3K4me3_Pro;E071,H3K4me3_Pro;E072,H3K4me3_Pro;E073,H3K4me3_Pro;E074,H3K4me3_Pro;E076,H3K4me3_Pro;E077,H3K4me3_Pro;E078,H3K4me3_Pro;E084,H3K4me3_Pro;E086,H3K4me3_Pro;E088,H3K4me3_Pro;E091,H3K4me3_Pro;E093,H3K4me3_Pro;E101,H3K4me3_Pro;E102,H3K4me3_Pro;E103,H3K4me3_Pro;E104,H3K4me3_Pro;E105,H3K4me3_Pro;E108,H3K4me3_Pro;E111,H3K4me3_Pro;E112,H3K4me3_Pro;E114,H3K4me3_Pro;E116,H3K4me3_Pro;E124,H3K4me3_Pro;E126,H3K4me3_Pro;E023,H3K9ac_Pro;E025,H3K9ac_Pro;E026,H3K9ac_Pro;E052,H3K9ac_Pro;E062,H3K9ac_Pro;E063,H3K9ac_Pro;E066,H3K9ac_Pro;E068,H3K9ac_Pro;E076,H3K9ac_Pro;E077,H3K9ac_Pro;E086,H3K9ac_Pro;E101,H3K9ac_Pro;E102,H3K9ac_Pro;E116,H3K9ac_Pro;E124,H3K9ac_Pro;E029,H3K27ac_Enh;E032,H3K27ac_Enh;E046,H3K27ac_Enh;E062,H3K27ac_Enh;E063,H3K27ac_Enh;E066,H3K27ac_Enh;E067,H3K27ac_Enh;E068,H3K27ac_Enh;E069,H3K27ac_Enh;E072,H3K27ac_Enh;E073,H3K27ac_Enh;E074,H3K27ac_Enh;E075,H3K27ac_Enh;E076,H3K27ac_Enh;E078,H3K27ac_Enh;E084,H3K27ac_Enh;E093,H3K27ac_Enh;E101,H3K27ac_Enh;E102,H3K27ac_Enh;E103,H3K27ac_Enh;E116,H3K27ac_Enh;E124,H3K27ac_Enh;E126,H3K27ac_Enh</td>
<td align="left"></td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,4.21086496303076e-07;GTEx2015_v6,Artery_Tibial,LRRK2,7.12887180198536e-12;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,1.24729820163706e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,5.97959499940652e-10;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,1.00796507632068e-05;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.08361107066017e-05;Westra2013,Whole_Blood,-,1.9330946764156724E-31;Westra2013,Whole_Blood,LRRK2,1.807765948958669E-5</td>
<td align="left">.</td>
<td align="left">.</td>
<td align="left">Barx1;Barx2;CDP_2;En-1_1;Hoxb6;Irf_known7;Ncx_1;Pou5f1_known2;Sox_18;Sox_3;Sox_9</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">1609</td>
<td align="left">.</td>
<td align="left">rs7294619</td>
<td align="left">BLD, GI, KID</td>
<td align="left">FAT, BLD, LIV, BRN, GI</td>
<td align="left">40617201</td>
<td align="left">40617202</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0.939999997615814</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.07617</td>
<td align="left">5</td>
<td align="left">T/C</td>
<td align="right">12</td>
<td align="right">40617202</td>
<td align="right">40617202</td>
<td align="right">1</td>
<td align="left">ENSG00000225342</td>
<td align="left">non_coding_transcript_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">regulatory_region_variant</td>
<td align="left">1000Genomes,Cited,Frequency,gnomAD,TOPMed</td>
</tr>
<tr class="odd">
<td align="left">rs7294619</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40617202</td>
<td align="right">0</td>
<td align="right">-0.1276</td>
<td align="right">0.0140</td>
<td align="right">0.8783</td>
<td align="right">0.1217</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="left">FALSE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">40.61720</td>
<td align="right">3</td>
<td align="left">TRUE</td>
<td align="right">0.75</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">-9.114286</td>
<td align="right">12</td>
<td align="left">40223400</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">0.11</td>
<td align="right">0.06</td>
<td align="right">0.00</td>
<td align="right">0.11</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E023,7_Enh;E025,7_Enh;E029,1_TssA;E030,2_TssAFlnk;E031,2_TssAFlnk;E032,1_TssA;E033,1_TssA;E035,1_TssA;E046,1_TssA;E050,7_Enh;E051,1_TssA;E062,1_TssA;E063,7_Enh;E066,7_Enh;E068,7_Enh;E074,7_Enh;E076,7_Enh;E078,1_TssA;E086,1_TssA;E103,7_Enh;E116,10_TssBiv;E124,1_TssA</td>
<td align="left">E009,23_PromBiv;E023,22_PromP;E025,22_PromP;E029,4_PromD2;E030,3_PromD1;E031,4_PromD2;E032,4_PromD2;E033,4_PromD2;E034,22_PromP;E035,4_PromD2;E036,4_PromD2;E037,22_PromP;E038,4_PromD2;E039,12_TxEnhW;E040,12_TxEnhW;E041,12_TxEnhW;E042,22_PromP;E043,4_PromD2;E044,4_PromD2;E045,12_TxEnhW;E046,4_PromD2;E047,4_PromD2;E048,22_PromP;E050,4_PromD2;E051,4_PromD2;E052,19_DNase;E053,22_PromP;E055,22_PromP;E057,22_PromP;E061,22_PromP;E062,4_PromD2;E063,22_PromP;E066,4_PromD2;E067,22_PromP;E068,22_PromP;E069,22_PromP;E070,22_PromP;E071,12_TxEnhW;E072,22_PromP;E073,22_PromP;E074,19_DNase;E075,22_PromP;E076,4_PromD2;E077,4_PromD2;E078,4_PromD2;E080,22_PromP;E081,22_PromP;E084,22_PromP;E085,22_PromP;E086,22_PromP;E088,22_PromP;E091,22_PromP;E092,22_PromP;E093,22_PromP;E100,22_PromP;E101,4_PromD2;E102,4_PromD2;E103,22_PromP;E106,22_PromP;E108,22_PromP;E109,22_PromP;E110,22_PromP;E111,16_EnhW1;E112,4_PromD2;E113,4_PromD2;E114,22_PromP;E115,23_PromBiv;E116,4_PromD2;E118,22_PromP;E119,22_PromP;E122,22_PromP;E124,3_PromD1</td>
<td align="left">E023,H3K4me1_Enh;E025,H3K4me1_Enh;E026,H3K4me1_Enh;E027,H3K4me1_Enh;E028,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E032,H3K4me1_Enh;E035,H3K4me1_Enh;E036,H3K4me1_Enh;E038,H3K4me1_Enh;E041,H3K4me1_Enh;E043,H3K4me1_Enh;E047,H3K4me1_Enh;E048,H3K4me1_Enh;E049,H3K4me1_Enh;E050,H3K4me1_Enh;E051,H3K4me1_Enh;E052,H3K4me1_Enh;E061,H3K4me1_Enh;E062,H3K4me1_Enh;E063,H3K4me1_Enh;E066,H3K4me1_Enh;E067,H3K4me1_Enh;E068,H3K4me1_Enh;E069,H3K4me1_Enh;E071,H3K4me1_Enh;E072,H3K4me1_Enh;E073,H3K4me1_Enh;E074,H3K4me1_Enh;E076,H3K4me1_Enh;E077,H3K4me1_Enh;E078,H3K4me1_Enh;E080,H3K4me1_Enh;E087,H3K4me1_Enh;E102,H3K4me1_Enh;E103,H3K4me1_Enh;E108,H3K4me1_Enh;E109,H3K4me1_Enh;E111,H3K4me1_Enh;E114,H3K4me1_Enh;E124,H3K4me1_Enh;E129,H3K4me1_Enh;E023,H3K4me3_Pro;E024,H3K4me3_Pro;E025,H3K4me3_Pro;E029,H3K4me3_Pro;E030,H3K4me3_Pro;E031,H3K4me3_Pro;E032,H3K4me3_Pro;E033,H3K4me3_Pro;E035,H3K4me3_Pro;E036,H3K4me3_Pro;E039,H3K4me3_Pro;E041,H3K4me3_Pro;E043,H3K4me3_Pro;E045,H3K4me3_Pro;E046,H3K4me3_Pro;E047,H3K4me3_Pro;E049,H3K4me3_Pro;E051,H3K4me3_Pro;E052,H3K4me3_Pro;E062,H3K4me3_Pro;E063,H3K4me3_Pro;E065,H3K4me3_Pro;E066,H3K4me3_Pro;E067,H3K4me3_Pro;E068,H3K4me3_Pro;E071,H3K4me3_Pro;E072,H3K4me3_Pro;E073,H3K4me3_Pro;E074,H3K4me3_Pro;E076,H3K4me3_Pro;E077,H3K4me3_Pro;E078,H3K4me3_Pro;E084,H3K4me3_Pro;E086,H3K4me3_Pro;E088,H3K4me3_Pro;E091,H3K4me3_Pro;E093,H3K4me3_Pro;E101,H3K4me3_Pro;E102,H3K4me3_Pro;E103,H3K4me3_Pro;E104,H3K4me3_Pro;E105,H3K4me3_Pro;E108,H3K4me3_Pro;E111,H3K4me3_Pro;E112,H3K4me3_Pro;E114,H3K4me3_Pro;E116,H3K4me3_Pro;E124,H3K4me3_Pro;E126,H3K4me3_Pro;E023,H3K9ac_Pro;E025,H3K9ac_Pro;E026,H3K9ac_Pro;E052,H3K9ac_Pro;E062,H3K9ac_Pro;E063,H3K9ac_Pro;E066,H3K9ac_Pro;E068,H3K9ac_Pro;E076,H3K9ac_Pro;E077,H3K9ac_Pro;E086,H3K9ac_Pro;E101,H3K9ac_Pro;E102,H3K9ac_Pro;E116,H3K9ac_Pro;E124,H3K9ac_Pro;E029,H3K27ac_Enh;E032,H3K27ac_Enh;E046,H3K27ac_Enh;E062,H3K27ac_Enh;E063,H3K27ac_Enh;E066,H3K27ac_Enh;E067,H3K27ac_Enh;E068,H3K27ac_Enh;E069,H3K27ac_Enh;E072,H3K27ac_Enh;E073,H3K27ac_Enh;E074,H3K27ac_Enh;E075,H3K27ac_Enh;E076,H3K27ac_Enh;E078,H3K27ac_Enh;E084,H3K27ac_Enh;E093,H3K27ac_Enh;E101,H3K27ac_Enh;E102,H3K27ac_Enh;E103,H3K27ac_Enh;E116,H3K27ac_Enh;E124,H3K27ac_Enh;E126,H3K27ac_Enh</td>
<td align="left"></td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,4.21086496303076e-07;GTEx2015_v6,Artery_Tibial,LRRK2,7.12887180198536e-12;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,1.24729820163706e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,5.97959499940652e-10;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,1.00796507632068e-05;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.08361107066017e-05;Westra2013,Whole_Blood,-,1.9330946764156724E-31;Westra2013,Whole_Blood,LRRK2,1.807765948958669E-5</td>
<td align="left">.</td>
<td align="left">.</td>
<td align="left">Barx1;Barx2;CDP_2;En-1_1;Hoxb6;Irf_known7;Ncx_1;Pou5f1_known2;Sox_18;Sox_3;Sox_9</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">1609</td>
<td align="left">.</td>
<td align="left">rs7294619</td>
<td align="left">BLD, GI, KID</td>
<td align="left">FAT, BLD, LIV, BRN, GI</td>
<td align="left">40617201</td>
<td align="left">40617202</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0.939999997615814</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.07617</td>
<td align="left">5</td>
<td align="left">T/C</td>
<td align="right">12</td>
<td align="right">40617202</td>
<td align="right">40617202</td>
<td align="right">1</td>
<td align="left">ENSG00000225342</td>
<td align="left">intron_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">regulatory_region_variant</td>
<td align="left">1000Genomes,Cited,Frequency,gnomAD,TOPMed</td>
</tr>
<tr class="even">
<td align="left">rs7294619</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40617202</td>
<td align="right">0</td>
<td align="right">-0.1276</td>
<td align="right">0.0140</td>
<td align="right">0.8783</td>
<td align="right">0.1217</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="left">FALSE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">40.61720</td>
<td align="right">3</td>
<td align="left">TRUE</td>
<td align="right">0.75</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">-9.114286</td>
<td align="right">12</td>
<td align="left">40223400</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">0.11</td>
<td align="right">0.06</td>
<td align="right">0.00</td>
<td align="right">0.11</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E023,7_Enh;E025,7_Enh;E029,1_TssA;E030,2_TssAFlnk;E031,2_TssAFlnk;E032,1_TssA;E033,1_TssA;E035,1_TssA;E046,1_TssA;E050,7_Enh;E051,1_TssA;E062,1_TssA;E063,7_Enh;E066,7_Enh;E068,7_Enh;E074,7_Enh;E076,7_Enh;E078,1_TssA;E086,1_TssA;E103,7_Enh;E116,10_TssBiv;E124,1_TssA</td>
<td align="left">E009,23_PromBiv;E023,22_PromP;E025,22_PromP;E029,4_PromD2;E030,3_PromD1;E031,4_PromD2;E032,4_PromD2;E033,4_PromD2;E034,22_PromP;E035,4_PromD2;E036,4_PromD2;E037,22_PromP;E038,4_PromD2;E039,12_TxEnhW;E040,12_TxEnhW;E041,12_TxEnhW;E042,22_PromP;E043,4_PromD2;E044,4_PromD2;E045,12_TxEnhW;E046,4_PromD2;E047,4_PromD2;E048,22_PromP;E050,4_PromD2;E051,4_PromD2;E052,19_DNase;E053,22_PromP;E055,22_PromP;E057,22_PromP;E061,22_PromP;E062,4_PromD2;E063,22_PromP;E066,4_PromD2;E067,22_PromP;E068,22_PromP;E069,22_PromP;E070,22_PromP;E071,12_TxEnhW;E072,22_PromP;E073,22_PromP;E074,19_DNase;E075,22_PromP;E076,4_PromD2;E077,4_PromD2;E078,4_PromD2;E080,22_PromP;E081,22_PromP;E084,22_PromP;E085,22_PromP;E086,22_PromP;E088,22_PromP;E091,22_PromP;E092,22_PromP;E093,22_PromP;E100,22_PromP;E101,4_PromD2;E102,4_PromD2;E103,22_PromP;E106,22_PromP;E108,22_PromP;E109,22_PromP;E110,22_PromP;E111,16_EnhW1;E112,4_PromD2;E113,4_PromD2;E114,22_PromP;E115,23_PromBiv;E116,4_PromD2;E118,22_PromP;E119,22_PromP;E122,22_PromP;E124,3_PromD1</td>
<td align="left">E023,H3K4me1_Enh;E025,H3K4me1_Enh;E026,H3K4me1_Enh;E027,H3K4me1_Enh;E028,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E032,H3K4me1_Enh;E035,H3K4me1_Enh;E036,H3K4me1_Enh;E038,H3K4me1_Enh;E041,H3K4me1_Enh;E043,H3K4me1_Enh;E047,H3K4me1_Enh;E048,H3K4me1_Enh;E049,H3K4me1_Enh;E050,H3K4me1_Enh;E051,H3K4me1_Enh;E052,H3K4me1_Enh;E061,H3K4me1_Enh;E062,H3K4me1_Enh;E063,H3K4me1_Enh;E066,H3K4me1_Enh;E067,H3K4me1_Enh;E068,H3K4me1_Enh;E069,H3K4me1_Enh;E071,H3K4me1_Enh;E072,H3K4me1_Enh;E073,H3K4me1_Enh;E074,H3K4me1_Enh;E076,H3K4me1_Enh;E077,H3K4me1_Enh;E078,H3K4me1_Enh;E080,H3K4me1_Enh;E087,H3K4me1_Enh;E102,H3K4me1_Enh;E103,H3K4me1_Enh;E108,H3K4me1_Enh;E109,H3K4me1_Enh;E111,H3K4me1_Enh;E114,H3K4me1_Enh;E124,H3K4me1_Enh;E129,H3K4me1_Enh;E023,H3K4me3_Pro;E024,H3K4me3_Pro;E025,H3K4me3_Pro;E029,H3K4me3_Pro;E030,H3K4me3_Pro;E031,H3K4me3_Pro;E032,H3K4me3_Pro;E033,H3K4me3_Pro;E035,H3K4me3_Pro;E036,H3K4me3_Pro;E039,H3K4me3_Pro;E041,H3K4me3_Pro;E043,H3K4me3_Pro;E045,H3K4me3_Pro;E046,H3K4me3_Pro;E047,H3K4me3_Pro;E049,H3K4me3_Pro;E051,H3K4me3_Pro;E052,H3K4me3_Pro;E062,H3K4me3_Pro;E063,H3K4me3_Pro;E065,H3K4me3_Pro;E066,H3K4me3_Pro;E067,H3K4me3_Pro;E068,H3K4me3_Pro;E071,H3K4me3_Pro;E072,H3K4me3_Pro;E073,H3K4me3_Pro;E074,H3K4me3_Pro;E076,H3K4me3_Pro;E077,H3K4me3_Pro;E078,H3K4me3_Pro;E084,H3K4me3_Pro;E086,H3K4me3_Pro;E088,H3K4me3_Pro;E091,H3K4me3_Pro;E093,H3K4me3_Pro;E101,H3K4me3_Pro;E102,H3K4me3_Pro;E103,H3K4me3_Pro;E104,H3K4me3_Pro;E105,H3K4me3_Pro;E108,H3K4me3_Pro;E111,H3K4me3_Pro;E112,H3K4me3_Pro;E114,H3K4me3_Pro;E116,H3K4me3_Pro;E124,H3K4me3_Pro;E126,H3K4me3_Pro;E023,H3K9ac_Pro;E025,H3K9ac_Pro;E026,H3K9ac_Pro;E052,H3K9ac_Pro;E062,H3K9ac_Pro;E063,H3K9ac_Pro;E066,H3K9ac_Pro;E068,H3K9ac_Pro;E076,H3K9ac_Pro;E077,H3K9ac_Pro;E086,H3K9ac_Pro;E101,H3K9ac_Pro;E102,H3K9ac_Pro;E116,H3K9ac_Pro;E124,H3K9ac_Pro;E029,H3K27ac_Enh;E032,H3K27ac_Enh;E046,H3K27ac_Enh;E062,H3K27ac_Enh;E063,H3K27ac_Enh;E066,H3K27ac_Enh;E067,H3K27ac_Enh;E068,H3K27ac_Enh;E069,H3K27ac_Enh;E072,H3K27ac_Enh;E073,H3K27ac_Enh;E074,H3K27ac_Enh;E075,H3K27ac_Enh;E076,H3K27ac_Enh;E078,H3K27ac_Enh;E084,H3K27ac_Enh;E093,H3K27ac_Enh;E101,H3K27ac_Enh;E102,H3K27ac_Enh;E103,H3K27ac_Enh;E116,H3K27ac_Enh;E124,H3K27ac_Enh;E126,H3K27ac_Enh</td>
<td align="left"></td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,4.21086496303076e-07;GTEx2015_v6,Artery_Tibial,LRRK2,7.12887180198536e-12;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,1.24729820163706e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,5.97959499940652e-10;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,1.00796507632068e-05;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.08361107066017e-05;Westra2013,Whole_Blood,-,1.9330946764156724E-31;Westra2013,Whole_Blood,LRRK2,1.807765948958669E-5</td>
<td align="left">.</td>
<td align="left">.</td>
<td align="left">Barx1;Barx2;CDP_2;En-1_1;Hoxb6;Irf_known7;Ncx_1;Pou5f1_known2;Sox_18;Sox_3;Sox_9</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">1609</td>
<td align="left">.</td>
<td align="left">rs7294619</td>
<td align="left">BLD, GI, KID</td>
<td align="left">FAT, BLD, LIV, BRN, GI</td>
<td align="left">40617201</td>
<td align="left">40617202</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0.939999997615814</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.07617</td>
<td align="left">5</td>
<td align="left">T/C</td>
<td align="right">12</td>
<td align="right">40617202</td>
<td align="right">40617202</td>
<td align="right">1</td>
<td align="left">ENSG00000188906</td>
<td align="left">intron_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">regulatory_region_variant</td>
<td align="left">1000Genomes,Cited,Frequency,gnomAD,TOPMed</td>
</tr>
<tr class="odd">
<td align="left">rs74324737</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40625081</td>
<td align="right">0</td>
<td align="right">0.1246</td>
<td align="right">0.0141</td>
<td align="right">0.1216</td>
<td align="right">0.1216</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="left">FALSE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">40.62508</td>
<td align="right">2</td>
<td align="left">TRUE</td>
<td align="right">0.50</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">8.836879</td>
<td align="right">12</td>
<td align="left">40231279</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">G</td>
<td align="left">A</td>
<td align="right">0.09</td>
<td align="right">0.06</td>
<td align="right">0.00</td>
<td align="right">0.11</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E029,6_EnhG;E030,6_EnhG;E031,7_Enh;E032,6_EnhG;E080,7_Enh;E124,6_EnhG</td>
<td align="left">E029,10_TxEnh5;E030,10_TxEnh5;E031,10_TxEnh5;E032,10_TxEnh5;E124,10_TxEnh5</td>
<td align="left">E029,H3K27ac_Enh;E032,H3K27ac_Enh;E046,H3K27ac_Enh;E068,H3K27ac_Enh;E080,H3K27ac_Enh;E124,H3K27ac_Enh;E029,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E032,H3K4me1_Enh;E067,H3K4me1_Enh;E073,H3K4me1_Enh;E078,H3K4me1_Enh;E116,H3K4me1_Enh;E124,H3K4me1_Enh;E031,H3K4me3_Pro;E067,H3K4me3_Pro;E116,H3K4me3_Pro;E124,H3K4me3_Pro;E124,H3K9ac_Pro</td>
<td align="left"></td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,7.71727072375557e-07;GTEx2015_v6,Artery_Tibial,LRRK2,3.79436480774191e-12;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,6.94075809560474e-06;GTEx2015_v6,Nerve_Tibial,LRRK2,8.74398504351068e-10;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,9.31129758632871e-06;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.18501232307152e-05</td>
<td align="left">.</td>
<td align="left">.</td>
<td align="left">E2A_3;E2A_5;Lmo2-complex_1;Mtf1_1;Myf_1;Myf_3;NF-E2_disc3;TCF12_disc1;ZEB1_known3</td>
<td align="left">ENSG00000188906.9</td>
<td align="left">LRRK2</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">INT</td>
<td align="left">rs74324737</td>
<td align="left"></td>
<td align="left">BLD, ADRL</td>
<td align="left">40625080</td>
<td align="left">40625081</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">1.79999995231628</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.9223</td>
<td align="left">5</td>
<td align="left">G/A</td>
<td align="right">12</td>
<td align="right">40625081</td>
<td align="right">40625081</td>
<td align="right">1</td>
<td align="left">ENSG00000188906</td>
<td align="left">intron_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
<td align="left">1000Genomes,Cited,Frequency,gnomAD,TOPMed</td>
</tr>
<tr class="even">
<td align="left">rs74324737</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40625081</td>
<td align="right">0</td>
<td align="right">0.1246</td>
<td align="right">0.0141</td>
<td align="right">0.1216</td>
<td align="right">0.1216</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="left">FALSE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">40.62508</td>
<td align="right">2</td>
<td align="left">TRUE</td>
<td align="right">0.50</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">8.836879</td>
<td align="right">12</td>
<td align="left">40231279</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">G</td>
<td align="left">A</td>
<td align="right">0.09</td>
<td align="right">0.06</td>
<td align="right">0.00</td>
<td align="right">0.11</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E029,6_EnhG;E030,6_EnhG;E031,7_Enh;E032,6_EnhG;E080,7_Enh;E124,6_EnhG</td>
<td align="left">E029,10_TxEnh5;E030,10_TxEnh5;E031,10_TxEnh5;E032,10_TxEnh5;E124,10_TxEnh5</td>
<td align="left">E029,H3K27ac_Enh;E032,H3K27ac_Enh;E046,H3K27ac_Enh;E068,H3K27ac_Enh;E080,H3K27ac_Enh;E124,H3K27ac_Enh;E029,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E032,H3K4me1_Enh;E067,H3K4me1_Enh;E073,H3K4me1_Enh;E078,H3K4me1_Enh;E116,H3K4me1_Enh;E124,H3K4me1_Enh;E031,H3K4me3_Pro;E067,H3K4me3_Pro;E116,H3K4me3_Pro;E124,H3K4me3_Pro;E124,H3K9ac_Pro</td>
<td align="left"></td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,7.71727072375557e-07;GTEx2015_v6,Artery_Tibial,LRRK2,3.79436480774191e-12;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,6.94075809560474e-06;GTEx2015_v6,Nerve_Tibial,LRRK2,8.74398504351068e-10;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,9.31129758632871e-06;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.18501232307152e-05</td>
<td align="left">.</td>
<td align="left">.</td>
<td align="left">E2A_3;E2A_5;Lmo2-complex_1;Mtf1_1;Myf_1;Myf_3;NF-E2_disc3;TCF12_disc1;ZEB1_known3</td>
<td align="left">ENSG00000188906.9</td>
<td align="left">LRRK2</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">INT</td>
<td align="left">rs74324737</td>
<td align="left"></td>
<td align="left">BLD, ADRL</td>
<td align="left">40625080</td>
<td align="left">40625081</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">1.79999995231628</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.9223</td>
<td align="left">5</td>
<td align="left">G/A</td>
<td align="right">12</td>
<td align="right">40625081</td>
<td align="right">40625081</td>
<td align="right">1</td>
<td align="left">ENSG00000188906</td>
<td align="left">non_coding_transcript_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
<td align="left">1000Genomes,Cited,Frequency,gnomAD,TOPMed</td>
</tr>
<tr class="odd">
<td align="left">rs76904798</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40614434</td>
<td align="right">0</td>
<td align="right">0.1439</td>
<td align="right">0.0130</td>
<td align="right">0.1444</td>
<td align="right">0.1444</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="left">TRUE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">40.61443</td>
<td align="right">3</td>
<td align="left">TRUE</td>
<td align="right">0.75</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">11.069231</td>
<td align="right">12</td>
<td align="left">40220632</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">0.11</td>
<td align="right">0.21</td>
<td align="right">0.03</td>
<td align="right">0.13</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E029,7_Enh;E030,7_Enh;E124,7_Enh</td>
<td align="left"></td>
<td align="left">E029,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E124,H3K4me1_Enh;E124,H3K27ac_Enh</td>
<td align="left"></td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,2.6136557196757e-09;GTEx2015_v6,Artery_Coronary,LRRK2,8.10828434519411e-08;GTEx2015_v6,Artery_Tibial,LRRK2,1.29589062525083e-14;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,2.01546552400868e-05;GTEx2015_v6,Esophagus_Muscularis,LRRK2,1.17034509288903e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,1.17833931025132e-15;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,3.02678179339592e-06;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.21658440247309e-07</td>
<td align="left">25064009,Parkinson’s disease,5E-14</td>
<td align="left">.</td>
<td align="left">Dbx1;Hmx_2;Hoxb8;Hoxd8;Ncx_2;Nkx6-1_3;Pou2f2_known2;Sox_14;Sox_15;Sox_16;Sox_18;Sox_19;Sox_2;Sox_5;Sox_7;Sox_9</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">4377</td>
<td align="left">.</td>
<td align="left">rs76904798</td>
<td align="left"></td>
<td align="left">BLD</td>
<td align="left">40614433</td>
<td align="left">40614434</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0.230000004172325</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.14464</td>
<td align="left">6</td>
<td align="left">C/T</td>
<td align="right">12</td>
<td align="right">40614434</td>
<td align="right">40614434</td>
<td align="right">1</td>
<td align="left">ENSG00000225342</td>
<td align="left">non_coding_transcript_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
<td align="left">1000Genomes,Cited,Frequency,gnomAD,Phenotype_or_Disease,TOPMed</td>
</tr>
<tr class="even">
<td align="left">rs76904798</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40614434</td>
<td align="right">0</td>
<td align="right">0.1439</td>
<td align="right">0.0130</td>
<td align="right">0.1444</td>
<td align="right">0.1444</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="left">TRUE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">40.61443</td>
<td align="right">3</td>
<td align="left">TRUE</td>
<td align="right">0.75</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">11.069231</td>
<td align="right">12</td>
<td align="left">40220632</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">0.11</td>
<td align="right">0.21</td>
<td align="right">0.03</td>
<td align="right">0.13</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E029,7_Enh;E030,7_Enh;E124,7_Enh</td>
<td align="left"></td>
<td align="left">E029,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E124,H3K4me1_Enh;E124,H3K27ac_Enh</td>
<td align="left"></td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,2.6136557196757e-09;GTEx2015_v6,Artery_Coronary,LRRK2,8.10828434519411e-08;GTEx2015_v6,Artery_Tibial,LRRK2,1.29589062525083e-14;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,2.01546552400868e-05;GTEx2015_v6,Esophagus_Muscularis,LRRK2,1.17034509288903e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,1.17833931025132e-15;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,3.02678179339592e-06;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.21658440247309e-07</td>
<td align="left">25064009,Parkinson’s disease,5E-14</td>
<td align="left">.</td>
<td align="left">Dbx1;Hmx_2;Hoxb8;Hoxd8;Ncx_2;Nkx6-1_3;Pou2f2_known2;Sox_14;Sox_15;Sox_16;Sox_18;Sox_19;Sox_2;Sox_5;Sox_7;Sox_9</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">4377</td>
<td align="left">.</td>
<td align="left">rs76904798</td>
<td align="left"></td>
<td align="left">BLD</td>
<td align="left">40614433</td>
<td align="left">40614434</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0.230000004172325</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.14464</td>
<td align="left">6</td>
<td align="left">C/T</td>
<td align="right">12</td>
<td align="right">40614434</td>
<td align="right">40614434</td>
<td align="right">1</td>
<td align="left">ENSG00000225342</td>
<td align="left">intron_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
<td align="left">1000Genomes,Cited,Frequency,gnomAD,Phenotype_or_Disease,TOPMed</td>
</tr>
<tr class="odd">
<td align="left">rs76904798</td>
<td align="left">LRRK2</td>
<td align="right">12</td>
<td align="right">40614434</td>
<td align="right">0</td>
<td align="right">0.1439</td>
<td align="right">0.0130</td>
<td align="right">0.1444</td>
<td align="right">0.1444</td>
<td align="right">56306</td>
<td align="right">1417791</td>
<td align="right">0.0382</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="left">TRUE</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">40.61443</td>
<td align="right">3</td>
<td align="left">TRUE</td>
<td align="right">0.75</td>
<td align="right">0</td>
<td align="right">216621</td>
<td align="right">11.069231</td>
<td align="right">12</td>
<td align="left">40220632</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">0.11</td>
<td align="right">0.21</td>
<td align="right">0.03</td>
<td align="right">0.13</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">E029,7_Enh;E030,7_Enh;E124,7_Enh</td>
<td align="left"></td>
<td align="left">E029,H3K4me1_Enh;E030,H3K4me1_Enh;E031,H3K4me1_Enh;E124,H3K4me1_Enh;E124,H3K27ac_Enh</td>
<td align="left"></td>
<td align="left">.</td>
<td align="left">GTEx2015_v6,Adipose_Subcutaneous,LRRK2,2.6136557196757e-09;GTEx2015_v6,Artery_Coronary,LRRK2,8.10828434519411e-08;GTEx2015_v6,Artery_Tibial,LRRK2,1.29589062525083e-14;GTEx2015_v6,Cells_Transformed_fibroblasts,LRRK2,2.01546552400868e-05;GTEx2015_v6,Esophagus_Muscularis,LRRK2,1.17034509288903e-05;GTEx2015_v6,Nerve_Tibial,LRRK2,1.17833931025132e-15;GTEx2015_v6,Skin_Not_Sun_Exposed_Suprapubic,LRRK2,3.02678179339592e-06;GTEx2015_v6,Skin_Sun_Exposed_Lower_leg,LRRK2,1.21658440247309e-07</td>
<td align="left">25064009,Parkinson’s disease,5E-14</td>
<td align="left">.</td>
<td align="left">Dbx1;Hmx_2;Hoxb8;Hoxd8;Ncx_2;Nkx6-1_3;Pou2f2_known2;Sox_14;Sox_15;Sox_16;Sox_18;Sox_19;Sox_2;Sox_5;Sox_7;Sox_9</td>
<td align="left">ENSG00000225342.1</td>
<td align="left">AC079630.4</td>
<td align="left">0</td>
<td align="left">0</td>
<td align="left">NM_198578</td>
<td align="left">LRRK2</td>
<td align="left">5</td>
<td align="left">4377</td>
<td align="left">.</td>
<td align="left">rs76904798</td>
<td align="left"></td>
<td align="left">BLD</td>
<td align="left">40614433</td>
<td align="left">40614434</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0</td>
<td align="left">0.230000004172325</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">0.14464</td>
<td align="left">6</td>
<td align="left">C/T</td>
<td align="right">12</td>
<td align="right">40614434</td>
<td align="right">40614434</td>
<td align="right">1</td>
<td align="left">ENSG00000188906</td>
<td align="left">intron_variant</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
<td align="left">1000Genomes,Cited,Frequency,gnomAD,Phenotype_or_Disease,TOPMed</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="summary-plots">Summary plots<a class="anchor" aria-label="anchor" href="#summary-plots"></a>
</h2>
<div class="section level3">
<h3 id="credible-set-bin-plot">Credible Set bin plot<a class="anchor" aria-label="anchor" href="#credible-set-bin-plot"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg_cs_bin</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/CS_bin_plot.html">CS_bin_plot</a></span><span class="op">(</span>merged_DT <span class="op">=</span> <span class="va">merged_DT</span>,</span>
<span>                                    show_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="credible-set-counts-plot">Credible Set counts plot<a class="anchor" aria-label="anchor" href="#credible-set-counts-plot"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg_cs_counts</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/CS_counts_plot.html">CS_counts_plot</a></span><span class="op">(</span>merged_DT <span class="op">=</span> <span class="va">merged_DT</span>, </span>
<span>                                          show_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="epigenomic-data">Epigenomic data<a class="anchor" aria-label="anchor" href="#epigenomic-data"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg_epi</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/peak_overlap_plot.html">peak_overlap_plot</a></span><span class="op">(</span></span>
<span>    merged_DT <span class="op">=</span> <span class="va">merged_DT</span>, </span>
<span>    include.NOTT2019_enhancers_promoters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    include.NOTT2019_PLACseq <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    <span class="co">#### Omit many annotations to save time ####</span></span>
<span>    include.NOTT2019_peaks <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    include.CORCES2020_scATACpeaks <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>    include.CORCES2020_Cicero_coaccess <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>    include.CORCES2020_bulkATACpeaks <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>    include.CORCES2020_HiChIP_FitHiChIP_coaccess <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    include.CORCES2020_gene_annotations <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ++ NOTT2019:: Getting regulatory regions data.</span></span></code></pre>
<pre><code><span><span class="co">## Importing Astrocyte enhancers ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Astrocyte promoters ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Neuronal enhancers ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Neuronal promoters ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Oligo enhancers ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Oligo promoters ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Microglia enhancers ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Microglia promoters ...</span></span></code></pre>
<pre><code><span><span class="co">## Converting dat to GRanges object.</span></span>
<span><span class="co">## Converting dat to GRanges object.</span></span></code></pre>
<pre><code><span><span class="co">## 48 query SNP(s) detected with reference overlap.</span></span></code></pre>
<pre><code><span><span class="co">## ++ NOTT2019:: Getting interaction anchors data.</span></span></code></pre>
<pre><code><span><span class="co">## Importing Microglia interactome ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Neuronal interactome ...</span></span></code></pre>
<pre><code><span><span class="co">## Importing Oligo interactome ...</span></span></code></pre>
<pre><code><span><span class="co">## Converting dat to GRanges object.</span></span></code></pre>
<pre><code><span><span class="co">## 52 query SNP(s) detected with reference overlap.</span></span></code></pre>
<pre><code><span><span class="co">## Converting dat to GRanges object.</span></span></code></pre>
<pre><code><span><span class="co">## 44 query SNP(s) detected with reference overlap.</span></span></code></pre>
<pre><code><span><span class="co">## Found more than one class "simpleUnit" in cache; using the first, from namespace 'hexbin'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'ggbio'</span></span></code></pre>
<p><img src="echoannot_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="super-summary-plot">Super summary plot<a class="anchor" aria-label="anchor" href="#super-summary-plot"></a>
</h3>
<p>Creates one big merged plot using the subfunctions above.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">super_plot</span> <span class="op">&lt;-</span> <span class="fu">echoannot</span><span class="fu">::</span><span class="fu"><a href="../reference/super_summary_plot.html">super_summary_plot</a></span><span class="op">(</span>merged_DT <span class="op">=</span> <span class="va">merged_DT</span>, </span>
<span>                                            plot_missense <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] echoannot_0.99.10 BiocStyle_2.25.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] utf8_1.2.2                  reticulate_1.26            </span></span>
<span><span class="co">##   [3] R.utils_2.12.0              RUnit_0.4.32               </span></span>
<span><span class="co">##   [5] tidyselect_1.2.0            RSQLite_2.2.18             </span></span>
<span><span class="co">##   [7] AnnotationDbi_1.59.1        htmlwidgets_1.5.4          </span></span>
<span><span class="co">##   [9] grid_4.2.1                  BiocParallel_1.31.15       </span></span>
<span><span class="co">##  [11] XGR_1.1.8                   munsell_0.5.0              </span></span>
<span><span class="co">##  [13] codetools_0.2-18            ragg_1.2.4                 </span></span>
<span><span class="co">##  [15] interp_1.1-3                DT_0.26                    </span></span>
<span><span class="co">##  [17] withr_2.5.0                 colorspace_2.0-3           </span></span>
<span><span class="co">##  [19] OrganismDbi_1.39.1          Biobase_2.57.3             </span></span>
<span><span class="co">##  [21] filelock_1.0.2              highr_0.9                  </span></span>
<span><span class="co">##  [23] knitr_1.40                  supraHex_1.35.0            </span></span>
<span><span class="co">##  [25] rstudioapi_0.14             stats4_4.2.1               </span></span>
<span><span class="co">##  [27] DescTools_0.99.47           labeling_0.4.2             </span></span>
<span><span class="co">##  [29] MatrixGenerics_1.9.1        GenomeInfoDbData_1.2.9     </span></span>
<span><span class="co">##  [31] farver_2.1.1                bit64_4.0.5                </span></span>
<span><span class="co">##  [33] echoconda_0.99.7            rprojroot_2.0.3            </span></span>
<span><span class="co">##  [35] basilisk_1.9.12             vctrs_0.5.0                </span></span>
<span><span class="co">##  [37] generics_0.1.3              xfun_0.34                  </span></span>
<span><span class="co">##  [39] biovizBase_1.45.0           BiocFileCache_2.5.2        </span></span>
<span><span class="co">##  [41] R6_2.5.1                    GenomeInfoDb_1.33.16       </span></span>
<span><span class="co">##  [43] RJSONIO_1.3-1.6             AnnotationFilter_1.21.0    </span></span>
<span><span class="co">##  [45] bitops_1.0-7                cachem_1.0.6               </span></span>
<span><span class="co">##  [47] reshape_0.8.9               DelayedArray_0.23.2        </span></span>
<span><span class="co">##  [49] assertthat_0.2.1            BiocIO_1.7.1               </span></span>
<span><span class="co">##  [51] scales_1.2.1                nnet_7.3-18                </span></span>
<span><span class="co">##  [53] rootSolve_1.8.2.3           gtable_0.3.1               </span></span>
<span><span class="co">##  [55] lmom_2.9                    ggbio_1.45.0               </span></span>
<span><span class="co">##  [57] ensembldb_2.21.5            rlang_1.0.6                </span></span>
<span><span class="co">##  [59] systemfonts_1.0.4           echodata_0.99.15           </span></span>
<span><span class="co">##  [61] splines_4.2.1               lazyeval_0.2.2             </span></span>
<span><span class="co">##  [63] rtracklayer_1.57.0          dichromat_2.0-0.1          </span></span>
<span><span class="co">##  [65] hexbin_1.28.2               checkmate_2.1.0            </span></span>
<span><span class="co">##  [67] BiocManager_1.30.19         yaml_2.3.6                 </span></span>
<span><span class="co">##  [69] reshape2_1.4.4              backports_1.4.1            </span></span>
<span><span class="co">##  [71] GenomicFeatures_1.49.8      ggnetwork_0.5.10           </span></span>
<span><span class="co">##  [73] Hmisc_4.7-1                 RBGL_1.73.0                </span></span>
<span><span class="co">##  [75] tools_4.2.1                 bookdown_0.29              </span></span>
<span><span class="co">##  [77] ggplot2_3.3.6               ellipsis_0.3.2             </span></span>
<span><span class="co">##  [79] jquerylib_0.1.4             RColorBrewer_1.1-3         </span></span>
<span><span class="co">##  [81] proxy_0.4-27                BiocGenerics_0.43.4        </span></span>
<span><span class="co">##  [83] Rcpp_1.0.9                  plyr_1.8.7                 </span></span>
<span><span class="co">##  [85] base64enc_0.1-3             progress_1.2.2             </span></span>
<span><span class="co">##  [87] zlibbioc_1.43.0             purrr_0.3.5                </span></span>
<span><span class="co">##  [89] RCurl_1.98-1.9              basilisk.utils_1.9.4       </span></span>
<span><span class="co">##  [91] prettyunits_1.1.1           rpart_4.1.16               </span></span>
<span><span class="co">##  [93] deldir_1.0-6                S4Vectors_0.35.4           </span></span>
<span><span class="co">##  [95] cluster_2.1.4               SummarizedExperiment_1.27.3</span></span>
<span><span class="co">##  [97] ggrepel_0.9.1               fs_1.5.2                   </span></span>
<span><span class="co">##  [99] crul_1.3                    magrittr_2.0.3             </span></span>
<span><span class="co">## [101] data.table_1.14.4           echotabix_0.99.8           </span></span>
<span><span class="co">## [103] dnet_1.1.7                  openxlsx_4.2.5.1           </span></span>
<span><span class="co">## [105] mvtnorm_1.1-3               ProtGenerics_1.29.1        </span></span>
<span><span class="co">## [107] matrixStats_0.62.0          patchwork_1.1.2            </span></span>
<span><span class="co">## [109] hms_1.1.2                   evaluate_0.17              </span></span>
<span><span class="co">## [111] XML_3.99-0.11               jpeg_0.1-9                 </span></span>
<span><span class="co">## [113] readxl_1.4.1                IRanges_2.31.2             </span></span>
<span><span class="co">## [115] gridExtra_2.3               compiler_4.2.1             </span></span>
<span><span class="co">## [117] biomaRt_2.53.3              tibble_3.1.8               </span></span>
<span><span class="co">## [119] haploR_4.0.6                crayon_1.5.2               </span></span>
<span><span class="co">## [121] R.oo_1.25.0                 htmltools_0.5.3            </span></span>
<span><span class="co">## [123] tzdb_0.3.0                  Formula_1.2-4              </span></span>
<span><span class="co">## [125] tidyr_1.2.1                 expm_0.999-6               </span></span>
<span><span class="co">## [127] Exact_3.2                   DBI_1.1.3                  </span></span>
<span><span class="co">## [129] dbplyr_2.2.1                MASS_7.3-58.1              </span></span>
<span><span class="co">## [131] rappdirs_0.3.3              boot_1.3-28                </span></span>
<span><span class="co">## [133] Matrix_1.5-1                readr_2.1.3                </span></span>
<span><span class="co">## [135] piggyback_0.1.4             cli_3.4.1                  </span></span>
<span><span class="co">## [137] R.methodsS3_1.8.2           parallel_4.2.1             </span></span>
<span><span class="co">## [139] igraph_1.3.5                GenomicRanges_1.49.1       </span></span>
<span><span class="co">## [141] pkgconfig_2.0.3             pkgdown_2.0.6.9000         </span></span>
<span><span class="co">## [143] GenomicAlignments_1.33.1    dir.expiry_1.5.1           </span></span>
<span><span class="co">## [145] RCircos_1.2.2               foreign_0.8-83             </span></span>
<span><span class="co">## [147] osfr_0.2.9                  xml2_1.3.3                 </span></span>
<span><span class="co">## [149] bslib_0.4.0                 XVector_0.37.1             </span></span>
<span><span class="co">## [151] stringr_1.4.1               VariantAnnotation_1.43.3   </span></span>
<span><span class="co">## [153] digest_0.6.30               graph_1.75.1               </span></span>
<span><span class="co">## [155] httpcode_0.3.0              Biostrings_2.65.6          </span></span>
<span><span class="co">## [157] rmarkdown_2.17              cellranger_1.1.0           </span></span>
<span><span class="co">## [159] htmlTable_2.4.1             gld_2.6.6                  </span></span>
<span><span class="co">## [161] restfulr_0.0.15             curl_4.3.3                 </span></span>
<span><span class="co">## [163] Rsamtools_2.13.4            rjson_0.2.21               </span></span>
<span><span class="co">## [165] lifecycle_1.0.3             nlme_3.1-160               </span></span>
<span><span class="co">## [167] jsonlite_1.8.3              viridisLite_0.4.1          </span></span>
<span><span class="co">## [169] desc_1.4.2                  BSgenome_1.65.4            </span></span>
<span><span class="co">## [171] fansi_1.0.3                 downloadR_0.99.5           </span></span>
<span><span class="co">## [173] pillar_1.8.1                lattice_0.20-45            </span></span>
<span><span class="co">## [175] GGally_2.1.2                KEGGREST_1.37.3            </span></span>
<span><span class="co">## [177] fastmap_1.1.0               httr_1.4.4                 </span></span>
<span><span class="co">## [179] survival_3.4-0              glue_1.6.2                 </span></span>
<span><span class="co">## [181] zip_2.2.2                   png_0.1-7                  </span></span>
<span><span class="co">## [183] bit_4.0.4                   Rgraphviz_2.41.2           </span></span>
<span><span class="co">## [185] class_7.3-20                stringi_1.7.8              </span></span>
<span><span class="co">## [187] sass_0.4.2                  blob_1.2.3                 </span></span>
<span><span class="co">## [189] textshaping_0.3.6           latticeExtra_0.6-30        </span></span>
<span><span class="co">## [191] memoise_2.0.1               dplyr_1.0.10               </span></span>
<span><span class="co">## [193] e1071_1.7-12                ape_5.6-2</span></span></code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
